// ionic2-calendar/dayview.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular 2 template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties}
 */
 /* tslint:disable */

import * as import0 from './dayview';
import * as import1 from '@angular/core/src/change_detection/change_detection';
import * as import2 from '@angular/core/src/linker/view';
import * as import3 from '@angular/core/src/linker/view_utils';
import * as import4 from '@angular/core/src/render/api';
import * as import5 from '@angular/core/src/metadata/view';
import * as import6 from '@angular/core/src/linker/view_type';
import * as import7 from '@angular/core/src/linker/component_factory';
import * as import8 from './calendar.service';
import * as import9 from '@angular/core/src/linker/query_list';
import * as import10 from 'ionic-angular/components/slides/slides';
import * as import11 from 'ionic-angular/components/slides/slides.ngfactory';
import * as import12 from 'ionic-angular/components/slides/slide';
import * as import13 from 'ionic-angular/components/slides/slide.ngfactory';
import * as import14 from 'ionic-angular/components/scroll/scroll';
import * as import15 from 'ionic-angular/components/scroll/scroll.ngfactory';
import * as import16 from '@angular/core/src/linker/view_container';
import * as import17 from '@angular/common/src/directives/ng_if.ngfactory';
import * as import18 from '@angular/common/src/pipes/date_pipe';
import * as import19 from 'ionic-angular/config/config';
import * as import20 from 'ionic-angular/platform/platform';
import * as import21 from '@angular/core/src/zone/ng_zone';
import * as import22 from 'ionic-angular/navigation/view-controller';
import * as import23 from '@angular/core/src/linker/element_ref';
import * as import24 from '@angular/core/src/linker/template_ref';
import * as import25 from '@angular/core/src/i18n/tokens';
import * as import26 from '@angular/common/src/directives/ng_if';
import * as import27 from '@angular/common/src/directives/ng_class.ngfactory';
import * as import28 from '@angular/common/src/directives/ng_style.ngfactory';
import * as import29 from '@angular/common/src/directives/ng_for.ngfactory';
import * as import30 from '@angular/core/src/change_detection/differs/iterable_differs';
import * as import31 from '@angular/core/src/change_detection/differs/keyvalue_differs';
import * as import32 from '@angular/common/src/directives/ng_for';
import * as import33 from '@angular/common/src/directives/ng_class';
import * as import34 from '@angular/common/src/directives/ng_style';
import * as import35 from '@angular/common/src/directives/ng_template_outlet.ngfactory';
import * as import36 from '@angular/common/src/directives/ng_template_outlet';
export class Wrapper_DayViewComponent {
  /*private*/ _eventHandler:Function;
  context:import0.DayViewComponent;
  /*private*/ _changed:boolean;
  /*private*/ _changes:{[key: string]:any};
  /*private*/ _expr_0:any;
  /*private*/ _expr_1:any;
  /*private*/ _expr_2:any;
  /*private*/ _expr_3:any;
  /*private*/ _expr_4:any;
  /*private*/ _expr_5:any;
  /*private*/ _expr_6:any;
  /*private*/ _expr_7:any;
  /*private*/ _expr_8:any;
  /*private*/ _expr_9:any;
  subscription0:any;
  subscription1:any;
  subscription2:any;
  subscription3:any;
  constructor(p0:any) {
    this._changed = false;
    this._changes = {};
    this.context = new import0.DayViewComponent(p0);
    this._expr_0 = import1.UNINITIALIZED;
    this._expr_1 = import1.UNINITIALIZED;
    this._expr_2 = import1.UNINITIALIZED;
    this._expr_3 = import1.UNINITIALIZED;
    this._expr_4 = import1.UNINITIALIZED;
    this._expr_5 = import1.UNINITIALIZED;
    this._expr_6 = import1.UNINITIALIZED;
    this._expr_7 = import1.UNINITIALIZED;
    this._expr_8 = import1.UNINITIALIZED;
    this._expr_9 = import1.UNINITIALIZED;
  }
  ngOnDetach(view:import2.AppView<any>,componentView:import2.AppView<any>,el:any):void {
  }
  ngOnDestroy():void {
    this.context.ngOnDestroy();
    (this.subscription0 && this.subscription0.unsubscribe());
    (this.subscription1 && this.subscription1.unsubscribe());
    (this.subscription2 && this.subscription2.unsubscribe());
    (this.subscription3 && this.subscription3.unsubscribe());
  }
  check_dayviewAllDayEventTemplate(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import3.checkBinding(throwOnChange,this._expr_0,currValue))) {
      this._changed = true;
      this.context.dayviewAllDayEventTemplate = currValue;
      this._changes['dayviewAllDayEventTemplate'] = new import1.SimpleChange(this._expr_0,currValue);
      this._expr_0 = currValue;
    }
  }
  check_dayviewNormalEventTemplate(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import3.checkBinding(throwOnChange,this._expr_1,currValue))) {
      this._changed = true;
      this.context.dayviewNormalEventTemplate = currValue;
      this._changes['dayviewNormalEventTemplate'] = new import1.SimpleChange(this._expr_1,currValue);
      this._expr_1 = currValue;
    }
  }
  check_formatHourColumn(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import3.checkBinding(throwOnChange,this._expr_2,currValue))) {
      this._changed = true;
      this.context.formatHourColumn = currValue;
      this._changes['formatHourColumn'] = new import1.SimpleChange(this._expr_2,currValue);
      this._expr_2 = currValue;
    }
  }
  check_formatDayTitle(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import3.checkBinding(throwOnChange,this._expr_3,currValue))) {
      this._changed = true;
      this.context.formatDayTitle = currValue;
      this._changes['formatDayTitle'] = new import1.SimpleChange(this._expr_3,currValue);
      this._expr_3 = currValue;
    }
  }
  check_allDayLabel(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import3.checkBinding(throwOnChange,this._expr_4,currValue))) {
      this._changed = true;
      this.context.allDayLabel = currValue;
      this._changes['allDayLabel'] = new import1.SimpleChange(this._expr_4,currValue);
      this._expr_4 = currValue;
    }
  }
  check_hourParts(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import3.checkBinding(throwOnChange,this._expr_5,currValue))) {
      this._changed = true;
      this.context.hourParts = currValue;
      this._changes['hourParts'] = new import1.SimpleChange(this._expr_5,currValue);
      this._expr_5 = currValue;
    }
  }
  check_eventSource(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import3.checkBinding(throwOnChange,this._expr_6,currValue))) {
      this._changed = true;
      this.context.eventSource = currValue;
      this._changes['eventSource'] = new import1.SimpleChange(this._expr_6,currValue);
      this._expr_6 = currValue;
    }
  }
  check_markDisabled(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import3.checkBinding(throwOnChange,this._expr_7,currValue))) {
      this._changed = true;
      this.context.markDisabled = currValue;
      this._changes['markDisabled'] = new import1.SimpleChange(this._expr_7,currValue);
      this._expr_7 = currValue;
    }
  }
  check_locale(currValue:any,throwOnChange:boolean,forceUpdate:boolean):void {
    if ((forceUpdate || import3.checkBinding(throwOnChange,this._expr_8,currValue))) {
      this._changed = true;
      this.context.locale = currValue;
      this._changes['locale'] = new import1.SimpleChange(this._expr_8,currValue);
      this._expr_8 = currValue;
    }
  }
  ngDoCheck(view:import2.AppView<any>,el:any,throwOnChange:boolean):boolean {
    var changed:any = this._changed;
    this._changed = false;
    if (!throwOnChange) {
      if (changed) {
        this.context.ngOnChanges(this._changes);
        this._changes = {};
      }
      if ((view.numberOfChecks === 0)) { this.context.ngOnInit(); }
    }
    return changed;
  }
  checkHost(view:import2.AppView<any>,componentView:import2.AppView<any>,el:any,throwOnChange:boolean):void {
    const currVal_9:any = this.context.class;
    if (import3.checkBinding(throwOnChange,this._expr_9,currVal_9)) {
      view.renderer.setElementClass(el,'dayview',currVal_9);
      this._expr_9 = currVal_9;
    }
  }
  handleEvent(eventName:string,$event:any):boolean {
    var result:boolean = true;
    return result;
  }
  subscribe(view:import2.AppView<any>,_eventHandler:any,emit0:boolean,emit1:boolean,emit2:boolean,emit3:boolean):void {
    this._eventHandler = _eventHandler;
    if (emit0) { (this.subscription0 = this.context.onRangeChanged.subscribe(_eventHandler.bind(view,'onRangeChanged'))); }
    if (emit1) { (this.subscription1 = this.context.onEventSelected.subscribe(_eventHandler.bind(view,'onEventSelected'))); }
    if (emit2) { (this.subscription2 = this.context.onTimeSelected.subscribe(_eventHandler.bind(view,'onTimeSelected'))); }
    if (emit3) { (this.subscription3 = this.context.onTitleChanged.subscribe(_eventHandler.bind(view,'onTitleChanged'))); }
  }
}
var renderType_DayViewComponent_Host:import4.RenderComponentType = import3.createRenderComponentType('',0,import5.ViewEncapsulation.None,([] as any[]),{});
class View_DayViewComponent_Host0 extends import2.AppView<any> {
  _el_0:any;
  compView_0:import2.AppView<import0.DayViewComponent>;
  _DayViewComponent_0_3:Wrapper_DayViewComponent;
  constructor(viewUtils:import3.ViewUtils,parentView:import2.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_DayViewComponent_Host0,renderType_DayViewComponent_Host,import6.ViewType.HOST,viewUtils,parentView,parentIndex,parentElement,import1.ChangeDetectorStatus.CheckAlways);
  }
  createInternal(rootSelector:string):import7.ComponentRef<any> {
    this._el_0 = import3.selectOrCreateRenderHostElement(this.renderer,'dayview',import3.EMPTY_INLINE_ARRAY,rootSelector,(null as any));
    this.compView_0 = new View_DayViewComponent0(this.viewUtils,this,0,this._el_0);
    this._DayViewComponent_0_3 = new Wrapper_DayViewComponent(this.injectorGet(import8.CalendarService,this.parentIndex));
    this.compView_0.create(this._DayViewComponent_0_3.context);
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [this._el_0]),(null as any));
    return new import7.ComponentRef_<any>(0,this,this._el_0,this._DayViewComponent_0_3.context);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import0.DayViewComponent) && (0 === requestNodeIndex))) { return this._DayViewComponent_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    this._DayViewComponent_0_3.ngDoCheck(this,this._el_0,throwOnChange);
    this._DayViewComponent_0_3.checkHost(this,this.compView_0,this._el_0,throwOnChange);
    this.compView_0.detectChanges(throwOnChange);
    if (!throwOnChange) { if ((this.numberOfChecks === 0)) { this._DayViewComponent_0_3.context.ngAfterViewInit(); } }
  }
  destroyInternal():void {
    this.compView_0.destroy();
    this._DayViewComponent_0_3.ngOnDestroy();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
export const DayViewComponentNgFactory:import7.ComponentFactory<import0.DayViewComponent> = new import7.ComponentFactory<import0.DayViewComponent>('dayview',View_DayViewComponent_Host0,import0.DayViewComponent);
const styles_DayViewComponent:any[] = ['\n        .table-fixed {\n          table-layout: fixed;\n        }\n\n        .table {\n          width: 100%;\n          max-width: 100%;\n          background-color: transparent;\n        }\n\n        .table > thead > tr > th, .table > tbody > tr > th, .table > tfoot > tr > th, .table > thead > tr > td,\n        .table > tbody > tr > td, .table > tfoot > tr > td {\n          padding: 8px;\n          line-height: 20px;\n          vertical-align: top;\n        }\n\n        .table > thead > tr > th {\n          vertical-align: bottom;\n          border-bottom: 2px solid #ddd;\n        }\n\n        .table > thead:first-child > tr:first-child > th, .table > thead:first-child > tr:first-child > td {\n          border-top: 0\n        }\n\n        .table > tbody + tbody {\n          border-top: 2px solid #ddd;\n        }\n\n        .table-bordered {\n          border: 1px solid #ddd;\n        }\n\n        .table-bordered > thead > tr > th, .table-bordered > tbody > tr > th, .table-bordered > tfoot > tr > th,\n        .table-bordered > thead > tr > td, .table-bordered > tbody > tr > td, .table-bordered > tfoot > tr > td {\n          border: 1px solid #ddd;\n        }\n\n        .table-bordered > thead > tr > th, .table-bordered > thead > tr > td {\n          border-bottom-width: 2px;\n        }\n\n        .table-striped > tbody > tr:nth-child(odd) > td, .table-striped > tbody > tr:nth-child(odd) > th {\n          background-color: #f9f9f9\n        }\n\n        .calendar-hour-column {\n          width: 50px;\n          white-space: nowrap;\n        }\n\n        .calendar-event-wrap {\n          position: relative;\n          width: 100%;\n          height: 100%;\n        }\n\n        .calendar-event {\n          position: absolute;\n          padding: 2px;\n          cursor: pointer;\n          z-index: 10000;\n        }\n\n        .calendar-cell {\n          padding: 0 !important;\n          height: 37px;\n        }\n\n        .dayview-allday-label {\n          float: left;\n          height: 100%;\n          line-height: 50px;\n          text-align: center;\n          width: 50px;\n        }\n\n        .dayview-allday-content-wrapper {\n          margin-left: 50px;\n          overflow: hidden;\n          height: 51px;\n        }\n\n        .dayview-allday-content-table {\n          min-height: 50px;\n        }\n\n        .dayview-allday-content-table td {\n          border-left: 1px solid #ddd;\n          border-right: 1px solid #ddd;\n        }\n\n        .dayview-allday-table {\n          height: 50px;\n          position: relative;\n          border-bottom: 1px solid #ddd;\n          font-size: 14px;\n        }\n\n        .dayview-normal-event-container {\n          margin-top: 50px;\n          overflow: hidden;\n          left: 0;\n          right: 0;\n          top: 0;\n          bottom: 0;\n          position: absolute;\n          font-size: 14px;\n        }\n\n        .dayview .slide-zoom {\n          height: 100%;\n        }\n\n        .dayview-allday-content-wrapper scroll-content {\n          width: 100%;\n        }\n\n        ::-webkit-scrollbar,\n        *::-webkit-scrollbar {\n          display: none;\n        }\n\n        .table > tbody > tr > td.calendar-hour-column {\n          padding-left: 0;\n          padding-right: 0;\n          vertical-align: middle;\n        }\n\n        @media (max-width: 750px) {\n          .dayview-allday-label, .calendar-hour-column {\n            width: 31px;\n            font-size: 12px;\n          }\n\n          .dayview-allday-label {\n            padding-top: 4px;\n          }\n\n          .table > tbody > tr > td.calendar-hour-column {\n            padding-left: 0;\n            padding-right: 0;\n            vertical-align: middle;\n            line-height: 12px;\n          }\n\n          .dayview-allday-label {\n            line-height: 20px;\n          }\n\n          .dayview-allday-content-wrapper {\n            margin-left: 31px;\n          }\n        }\n    '];
var renderType_DayViewComponent:import4.RenderComponentType = import3.createRenderComponentType('',0,import5.ViewEncapsulation.None,styles_DayViewComponent,{});
export class View_DayViewComponent0 extends import2.AppView<import0.DayViewComponent> {
  _viewQuery_daySlider_0:import9.QueryList<any>;
  _text_0:any;
  _el_1:any;
  compView_1:import2.AppView<import10.Slides>;
  _Slides_1_3:import11.Wrapper_Slides;
  _text_2:any;
  _el_3:any;
  compView_3:import2.AppView<import12.Slide>;
  _Slide_3_3:import13.Wrapper_Slide;
  _text_4:any;
  _el_5:any;
  _text_6:any;
  _el_7:any;
  _text_8:any;
  _text_9:any;
  _el_10:any;
  compView_10:import2.AppView<import14.Scroll>;
  _Scroll_10_3:import15.Wrapper_Scroll;
  _text_11:any;
  _el_12:any;
  _text_13:any;
  _el_14:any;
  _text_15:any;
  _el_16:any;
  _text_17:any;
  _anchor_18:any;
  /*private*/ _vc_18:import16.ViewContainer;
  _TemplateRef_18_5:any;
  _NgIf_18_6:import17.Wrapper_NgIf;
  _text_19:any;
  _anchor_20:any;
  /*private*/ _vc_20:import16.ViewContainer;
  _TemplateRef_20_5:any;
  _NgIf_20_6:import17.Wrapper_NgIf;
  _text_21:any;
  _text_22:any;
  _text_23:any;
  _text_24:any;
  _text_25:any;
  _text_26:any;
  _el_27:any;
  compView_27:import2.AppView<import14.Scroll>;
  _Scroll_27_3:import15.Wrapper_Scroll;
  _text_28:any;
  _anchor_29:any;
  /*private*/ _vc_29:import16.ViewContainer;
  _TemplateRef_29_5:any;
  _NgIf_29_6:import17.Wrapper_NgIf;
  _text_30:any;
  _anchor_31:any;
  /*private*/ _vc_31:import16.ViewContainer;
  _TemplateRef_31_5:any;
  _NgIf_31_6:import17.Wrapper_NgIf;
  _text_32:any;
  _text_33:any;
  _text_34:any;
  _el_35:any;
  compView_35:import2.AppView<import12.Slide>;
  _Slide_35_3:import13.Wrapper_Slide;
  _text_36:any;
  _el_37:any;
  _text_38:any;
  _el_39:any;
  _text_40:any;
  _text_41:any;
  _el_42:any;
  compView_42:import2.AppView<import14.Scroll>;
  _Scroll_42_3:import15.Wrapper_Scroll;
  _text_43:any;
  _el_44:any;
  _text_45:any;
  _el_46:any;
  _text_47:any;
  _el_48:any;
  _text_49:any;
  _anchor_50:any;
  /*private*/ _vc_50:import16.ViewContainer;
  _TemplateRef_50_5:any;
  _NgIf_50_6:import17.Wrapper_NgIf;
  _text_51:any;
  _anchor_52:any;
  /*private*/ _vc_52:import16.ViewContainer;
  _TemplateRef_52_5:any;
  _NgIf_52_6:import17.Wrapper_NgIf;
  _text_53:any;
  _text_54:any;
  _text_55:any;
  _text_56:any;
  _text_57:any;
  _text_58:any;
  _el_59:any;
  compView_59:import2.AppView<import14.Scroll>;
  _Scroll_59_3:import15.Wrapper_Scroll;
  _text_60:any;
  _anchor_61:any;
  /*private*/ _vc_61:import16.ViewContainer;
  _TemplateRef_61_5:any;
  _NgIf_61_6:import17.Wrapper_NgIf;
  _text_62:any;
  _anchor_63:any;
  /*private*/ _vc_63:import16.ViewContainer;
  _TemplateRef_63_5:any;
  _NgIf_63_6:import17.Wrapper_NgIf;
  _text_64:any;
  _text_65:any;
  _text_66:any;
  _el_67:any;
  compView_67:import2.AppView<import12.Slide>;
  _Slide_67_3:import13.Wrapper_Slide;
  _text_68:any;
  _el_69:any;
  _text_70:any;
  _el_71:any;
  _text_72:any;
  _text_73:any;
  _el_74:any;
  compView_74:import2.AppView<import14.Scroll>;
  _Scroll_74_3:import15.Wrapper_Scroll;
  _text_75:any;
  _el_76:any;
  _text_77:any;
  _el_78:any;
  _text_79:any;
  _el_80:any;
  _text_81:any;
  _anchor_82:any;
  /*private*/ _vc_82:import16.ViewContainer;
  _TemplateRef_82_5:any;
  _NgIf_82_6:import17.Wrapper_NgIf;
  _text_83:any;
  _anchor_84:any;
  /*private*/ _vc_84:import16.ViewContainer;
  _TemplateRef_84_5:any;
  _NgIf_84_6:import17.Wrapper_NgIf;
  _text_85:any;
  _text_86:any;
  _text_87:any;
  _text_88:any;
  _text_89:any;
  _text_90:any;
  _el_91:any;
  compView_91:import2.AppView<import14.Scroll>;
  _Scroll_91_3:import15.Wrapper_Scroll;
  _text_92:any;
  _anchor_93:any;
  /*private*/ _vc_93:import16.ViewContainer;
  _TemplateRef_93_5:any;
  _NgIf_93_6:import17.Wrapper_NgIf;
  _text_94:any;
  _anchor_95:any;
  /*private*/ _vc_95:import16.ViewContainer;
  _TemplateRef_95_5:any;
  _NgIf_95_6:import17.Wrapper_NgIf;
  _text_96:any;
  _text_97:any;
  _text_98:any;
  _text_99:any;
  /*private*/ _expr_157:any;
  _pipe_date_0:import18.DatePipe;
  /*private*/ _expr_159:any;
  /*private*/ _expr_160:any;
  constructor(viewUtils:import3.ViewUtils,parentView:import2.AppView<any>,parentIndex:number,parentElement:any) {
    super(View_DayViewComponent0,renderType_DayViewComponent,import6.ViewType.COMPONENT,viewUtils,parentView,parentIndex,parentElement,import1.ChangeDetectorStatus.CheckAlways);
    this._expr_157 = import1.UNINITIALIZED;
    this._expr_159 = import1.UNINITIALIZED;
    this._expr_160 = import1.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import7.ComponentRef<any> {
    const parentRenderNode:any = this.renderer.createViewRoot(this.parentElement);
    this._viewQuery_daySlider_0 = new import9.QueryList<any>();
    this._text_0 = this.renderer.createText(parentRenderNode,'\n        ',(null as any));
    this._el_1 = import3.createRenderElement(this.renderer,parentRenderNode,'ion-slides',import3.EMPTY_INLINE_ARRAY,(null as any));
    this.compView_1 = new import11.View_Slides0(this.viewUtils,this,1,this._el_1);
    this._Slides_1_3 = new import11.Wrapper_Slides(this.parentView.injectorGet(import19.Config,this.parentIndex),this.parentView.injectorGet(import20.Platform,this.parentIndex),this.parentView.injectorGet(import21.NgZone,this.parentIndex),this.parentView.injectorGet(import22.ViewController,this.parentIndex,(null as any)),new import23.ElementRef(this._el_1),this.renderer);
    this._text_2 = this.renderer.createText((null as any),'\n            ',(null as any));
    this._el_3 = import3.createRenderElement(this.renderer,(null as any),'ion-slide',import3.EMPTY_INLINE_ARRAY,(null as any));
    this.compView_3 = new import13.View_Slide0(this.viewUtils,this,3,this._el_3);
    this._Slide_3_3 = new import13.Wrapper_Slide(new import23.ElementRef(this._el_3),this.renderer,this._Slides_1_3.context);
    this._text_4 = this.renderer.createText((null as any),'\n                ',(null as any));
    this._el_5 = import3.createRenderElement(this.renderer,(null as any),'div',new import3.InlineArray2(2,'class','dayview-allday-table'),(null as any));
    this._text_6 = this.renderer.createText(this._el_5,'\n                    ',(null as any));
    this._el_7 = import3.createRenderElement(this.renderer,this._el_5,'div',new import3.InlineArray2(2,'class','dayview-allday-label'),(null as any));
    this._text_8 = this.renderer.createText(this._el_7,'',(null as any));
    this._text_9 = this.renderer.createText(this._el_5,'\n                    ',(null as any));
    this._el_10 = import3.createRenderElement(this.renderer,this._el_5,'ion-scroll',new import3.InlineArray8(6,'class','dayview-allday-content-wrapper','scrollY','true','zoom','false'),(null as any));
    this.compView_10 = new import15.View_Scroll0(this.viewUtils,this,10,this._el_10);
    this._Scroll_10_3 = new import15.Wrapper_Scroll(new import23.ElementRef(this._el_10));
    this._text_11 = this.renderer.createText((null as any),'\n                        ',(null as any));
    this._el_12 = import3.createRenderElement(this.renderer,(null as any),'table',new import3.InlineArray2(2,'class','table table-bordered dayview-allday-content-table'),(null as any));
    this._text_13 = this.renderer.createText(this._el_12,'\n                            ',(null as any));
    this._el_14 = import3.createRenderElement(this.renderer,this._el_12,'tbody',import3.EMPTY_INLINE_ARRAY,(null as any));
    this._text_15 = this.renderer.createText(this._el_14,'\n                            ',(null as any));
    this._el_16 = import3.createRenderElement(this.renderer,this._el_14,'tr',import3.EMPTY_INLINE_ARRAY,(null as any));
    this._text_17 = this.renderer.createText(this._el_16,'\n                                ',(null as any));
    this._anchor_18 = this.renderer.createTemplateAnchor(this._el_16,(null as any));
    this._vc_18 = new import16.ViewContainer(18,16,this,this._anchor_18);
    this._TemplateRef_18_5 = new import24.TemplateRef_(this,18,this._anchor_18);
    this._NgIf_18_6 = new import17.Wrapper_NgIf(this._vc_18.vcRef,this._TemplateRef_18_5);
    this._text_19 = this.renderer.createText(this._el_16,'\n                                ',(null as any));
    this._anchor_20 = this.renderer.createTemplateAnchor(this._el_16,(null as any));
    this._vc_20 = new import16.ViewContainer(20,16,this,this._anchor_20);
    this._TemplateRef_20_5 = new import24.TemplateRef_(this,20,this._anchor_20);
    this._NgIf_20_6 = new import17.Wrapper_NgIf(this._vc_20.vcRef,this._TemplateRef_20_5);
    this._text_21 = this.renderer.createText(this._el_16,'\n                            ',(null as any));
    this._text_22 = this.renderer.createText(this._el_14,'\n                            ',(null as any));
    this._text_23 = this.renderer.createText(this._el_12,'\n                        ',(null as any));
    this._text_24 = this.renderer.createText((null as any),'\n                    ',(null as any));
    this.compView_10.create(this._Scroll_10_3.context);
    this._text_25 = this.renderer.createText(this._el_5,'\n                ',(null as any));
    this._text_26 = this.renderer.createText((null as any),'\n                ',(null as any));
    this._el_27 = import3.createRenderElement(this.renderer,(null as any),'ion-scroll',new import3.InlineArray4(4,'class','dayview-normal-event-container','scrollY','true'),(null as any));
    this.compView_27 = new import15.View_Scroll0(this.viewUtils,this,27,this._el_27);
    this._Scroll_27_3 = new import15.Wrapper_Scroll(new import23.ElementRef(this._el_27));
    this._text_28 = this.renderer.createText((null as any),'\n                    ',(null as any));
    this._anchor_29 = this.renderer.createTemplateAnchor((null as any),(null as any));
    this._vc_29 = new import16.ViewContainer(29,27,this,this._anchor_29);
    this._TemplateRef_29_5 = new import24.TemplateRef_(this,29,this._anchor_29);
    this._NgIf_29_6 = new import17.Wrapper_NgIf(this._vc_29.vcRef,this._TemplateRef_29_5);
    this._text_30 = this.renderer.createText((null as any),'\n                    ',(null as any));
    this._anchor_31 = this.renderer.createTemplateAnchor((null as any),(null as any));
    this._vc_31 = new import16.ViewContainer(31,27,this,this._anchor_31);
    this._TemplateRef_31_5 = new import24.TemplateRef_(this,31,this._anchor_31);
    this._NgIf_31_6 = new import17.Wrapper_NgIf(this._vc_31.vcRef,this._TemplateRef_31_5);
    this._text_32 = this.renderer.createText((null as any),'\n                ',(null as any));
    this.compView_27.create(this._Scroll_27_3.context);
    this._text_33 = this.renderer.createText((null as any),'\n            ',(null as any));
    this.compView_3.create(this._Slide_3_3.context);
    this._text_34 = this.renderer.createText((null as any),'\n            ',(null as any));
    this._el_35 = import3.createRenderElement(this.renderer,(null as any),'ion-slide',import3.EMPTY_INLINE_ARRAY,(null as any));
    this.compView_35 = new import13.View_Slide0(this.viewUtils,this,35,this._el_35);
    this._Slide_35_3 = new import13.Wrapper_Slide(new import23.ElementRef(this._el_35),this.renderer,this._Slides_1_3.context);
    this._text_36 = this.renderer.createText((null as any),'\n                ',(null as any));
    this._el_37 = import3.createRenderElement(this.renderer,(null as any),'div',new import3.InlineArray2(2,'class','dayview-allday-table'),(null as any));
    this._text_38 = this.renderer.createText(this._el_37,'\n                    ',(null as any));
    this._el_39 = import3.createRenderElement(this.renderer,this._el_37,'div',new import3.InlineArray2(2,'class','dayview-allday-label'),(null as any));
    this._text_40 = this.renderer.createText(this._el_39,'',(null as any));
    this._text_41 = this.renderer.createText(this._el_37,'\n                    ',(null as any));
    this._el_42 = import3.createRenderElement(this.renderer,this._el_37,'ion-scroll',new import3.InlineArray8(6,'class','dayview-allday-content-wrapper','scrollY','true','zoom','false'),(null as any));
    this.compView_42 = new import15.View_Scroll0(this.viewUtils,this,42,this._el_42);
    this._Scroll_42_3 = new import15.Wrapper_Scroll(new import23.ElementRef(this._el_42));
    this._text_43 = this.renderer.createText((null as any),'\n                        ',(null as any));
    this._el_44 = import3.createRenderElement(this.renderer,(null as any),'table',new import3.InlineArray2(2,'class','table table-bordered dayview-allday-content-table'),(null as any));
    this._text_45 = this.renderer.createText(this._el_44,'\n                            ',(null as any));
    this._el_46 = import3.createRenderElement(this.renderer,this._el_44,'tbody',import3.EMPTY_INLINE_ARRAY,(null as any));
    this._text_47 = this.renderer.createText(this._el_46,'\n                            ',(null as any));
    this._el_48 = import3.createRenderElement(this.renderer,this._el_46,'tr',import3.EMPTY_INLINE_ARRAY,(null as any));
    this._text_49 = this.renderer.createText(this._el_48,'\n                                ',(null as any));
    this._anchor_50 = this.renderer.createTemplateAnchor(this._el_48,(null as any));
    this._vc_50 = new import16.ViewContainer(50,48,this,this._anchor_50);
    this._TemplateRef_50_5 = new import24.TemplateRef_(this,50,this._anchor_50);
    this._NgIf_50_6 = new import17.Wrapper_NgIf(this._vc_50.vcRef,this._TemplateRef_50_5);
    this._text_51 = this.renderer.createText(this._el_48,'\n                                ',(null as any));
    this._anchor_52 = this.renderer.createTemplateAnchor(this._el_48,(null as any));
    this._vc_52 = new import16.ViewContainer(52,48,this,this._anchor_52);
    this._TemplateRef_52_5 = new import24.TemplateRef_(this,52,this._anchor_52);
    this._NgIf_52_6 = new import17.Wrapper_NgIf(this._vc_52.vcRef,this._TemplateRef_52_5);
    this._text_53 = this.renderer.createText(this._el_48,'\n                            ',(null as any));
    this._text_54 = this.renderer.createText(this._el_46,'\n                            ',(null as any));
    this._text_55 = this.renderer.createText(this._el_44,'\n                        ',(null as any));
    this._text_56 = this.renderer.createText((null as any),'\n                    ',(null as any));
    this.compView_42.create(this._Scroll_42_3.context);
    this._text_57 = this.renderer.createText(this._el_37,'\n                ',(null as any));
    this._text_58 = this.renderer.createText((null as any),'\n                ',(null as any));
    this._el_59 = import3.createRenderElement(this.renderer,(null as any),'ion-scroll',new import3.InlineArray4(4,'class','dayview-normal-event-container','scrollY','true'),(null as any));
    this.compView_59 = new import15.View_Scroll0(this.viewUtils,this,59,this._el_59);
    this._Scroll_59_3 = new import15.Wrapper_Scroll(new import23.ElementRef(this._el_59));
    this._text_60 = this.renderer.createText((null as any),'\n                    ',(null as any));
    this._anchor_61 = this.renderer.createTemplateAnchor((null as any),(null as any));
    this._vc_61 = new import16.ViewContainer(61,59,this,this._anchor_61);
    this._TemplateRef_61_5 = new import24.TemplateRef_(this,61,this._anchor_61);
    this._NgIf_61_6 = new import17.Wrapper_NgIf(this._vc_61.vcRef,this._TemplateRef_61_5);
    this._text_62 = this.renderer.createText((null as any),'\n                    ',(null as any));
    this._anchor_63 = this.renderer.createTemplateAnchor((null as any),(null as any));
    this._vc_63 = new import16.ViewContainer(63,59,this,this._anchor_63);
    this._TemplateRef_63_5 = new import24.TemplateRef_(this,63,this._anchor_63);
    this._NgIf_63_6 = new import17.Wrapper_NgIf(this._vc_63.vcRef,this._TemplateRef_63_5);
    this._text_64 = this.renderer.createText((null as any),'\n                ',(null as any));
    this.compView_59.create(this._Scroll_59_3.context);
    this._text_65 = this.renderer.createText((null as any),'\n            ',(null as any));
    this.compView_35.create(this._Slide_35_3.context);
    this._text_66 = this.renderer.createText((null as any),'\n            ',(null as any));
    this._el_67 = import3.createRenderElement(this.renderer,(null as any),'ion-slide',import3.EMPTY_INLINE_ARRAY,(null as any));
    this.compView_67 = new import13.View_Slide0(this.viewUtils,this,67,this._el_67);
    this._Slide_67_3 = new import13.Wrapper_Slide(new import23.ElementRef(this._el_67),this.renderer,this._Slides_1_3.context);
    this._text_68 = this.renderer.createText((null as any),'\n                ',(null as any));
    this._el_69 = import3.createRenderElement(this.renderer,(null as any),'div',new import3.InlineArray2(2,'class','dayview-allday-table'),(null as any));
    this._text_70 = this.renderer.createText(this._el_69,'\n                    ',(null as any));
    this._el_71 = import3.createRenderElement(this.renderer,this._el_69,'div',new import3.InlineArray2(2,'class','dayview-allday-label'),(null as any));
    this._text_72 = this.renderer.createText(this._el_71,'',(null as any));
    this._text_73 = this.renderer.createText(this._el_69,'\n                    ',(null as any));
    this._el_74 = import3.createRenderElement(this.renderer,this._el_69,'ion-scroll',new import3.InlineArray8(6,'class','dayview-allday-content-wrapper','scrollY','true','zoom','false'),(null as any));
    this.compView_74 = new import15.View_Scroll0(this.viewUtils,this,74,this._el_74);
    this._Scroll_74_3 = new import15.Wrapper_Scroll(new import23.ElementRef(this._el_74));
    this._text_75 = this.renderer.createText((null as any),'\n                        ',(null as any));
    this._el_76 = import3.createRenderElement(this.renderer,(null as any),'table',new import3.InlineArray2(2,'class','table table-bordered dayview-allday-content-table'),(null as any));
    this._text_77 = this.renderer.createText(this._el_76,'\n                            ',(null as any));
    this._el_78 = import3.createRenderElement(this.renderer,this._el_76,'tbody',import3.EMPTY_INLINE_ARRAY,(null as any));
    this._text_79 = this.renderer.createText(this._el_78,'\n                            ',(null as any));
    this._el_80 = import3.createRenderElement(this.renderer,this._el_78,'tr',import3.EMPTY_INLINE_ARRAY,(null as any));
    this._text_81 = this.renderer.createText(this._el_80,'\n                                ',(null as any));
    this._anchor_82 = this.renderer.createTemplateAnchor(this._el_80,(null as any));
    this._vc_82 = new import16.ViewContainer(82,80,this,this._anchor_82);
    this._TemplateRef_82_5 = new import24.TemplateRef_(this,82,this._anchor_82);
    this._NgIf_82_6 = new import17.Wrapper_NgIf(this._vc_82.vcRef,this._TemplateRef_82_5);
    this._text_83 = this.renderer.createText(this._el_80,'\n                                ',(null as any));
    this._anchor_84 = this.renderer.createTemplateAnchor(this._el_80,(null as any));
    this._vc_84 = new import16.ViewContainer(84,80,this,this._anchor_84);
    this._TemplateRef_84_5 = new import24.TemplateRef_(this,84,this._anchor_84);
    this._NgIf_84_6 = new import17.Wrapper_NgIf(this._vc_84.vcRef,this._TemplateRef_84_5);
    this._text_85 = this.renderer.createText(this._el_80,'\n                            ',(null as any));
    this._text_86 = this.renderer.createText(this._el_78,'\n                            ',(null as any));
    this._text_87 = this.renderer.createText(this._el_76,'\n                        ',(null as any));
    this._text_88 = this.renderer.createText((null as any),'\n                    ',(null as any));
    this.compView_74.create(this._Scroll_74_3.context);
    this._text_89 = this.renderer.createText(this._el_69,'\n                ',(null as any));
    this._text_90 = this.renderer.createText((null as any),'\n                ',(null as any));
    this._el_91 = import3.createRenderElement(this.renderer,(null as any),'ion-scroll',new import3.InlineArray4(4,'class','dayview-normal-event-container','scrollY','true'),(null as any));
    this.compView_91 = new import15.View_Scroll0(this.viewUtils,this,91,this._el_91);
    this._Scroll_91_3 = new import15.Wrapper_Scroll(new import23.ElementRef(this._el_91));
    this._text_92 = this.renderer.createText((null as any),'\n                    ',(null as any));
    this._anchor_93 = this.renderer.createTemplateAnchor((null as any),(null as any));
    this._vc_93 = new import16.ViewContainer(93,91,this,this._anchor_93);
    this._TemplateRef_93_5 = new import24.TemplateRef_(this,93,this._anchor_93);
    this._NgIf_93_6 = new import17.Wrapper_NgIf(this._vc_93.vcRef,this._TemplateRef_93_5);
    this._text_94 = this.renderer.createText((null as any),'\n                    ',(null as any));
    this._anchor_95 = this.renderer.createTemplateAnchor((null as any),(null as any));
    this._vc_95 = new import16.ViewContainer(95,91,this,this._anchor_95);
    this._TemplateRef_95_5 = new import24.TemplateRef_(this,95,this._anchor_95);
    this._NgIf_95_6 = new import17.Wrapper_NgIf(this._vc_95.vcRef,this._TemplateRef_95_5);
    this._text_96 = this.renderer.createText((null as any),'\n                ',(null as any));
    this.compView_91.create(this._Scroll_91_3.context);
    this._text_97 = this.renderer.createText((null as any),'\n            ',(null as any));
    this.compView_67.create(this._Slide_67_3.context);
    this._text_98 = this.renderer.createText((null as any),'\n        ',(null as any));
    this.compView_1.create(this._Slides_1_3.context);
    this._text_99 = this.renderer.createText(parentRenderNode,'\n    ',(null as any));
    var disposable_0:Function = import3.subscribeToRenderElement(this,this._el_1,new import3.InlineArray2(2,'ionSlideDidChange',(null as any)),this.eventHandler(this.handleEvent_1));
    this._Slides_1_3.subscribe(this,this.eventHandler(this.handleEvent_1),false,true,false,false,false,false,false,false,false,false,false,false,false,false,false,false,false);
    this._pipe_date_0 = new import18.DatePipe(this.parentView.injectorGet(import25.LOCALE_ID,this.parentIndex));
    this._viewQuery_daySlider_0.reset([this._Slides_1_3.context]);
    this.context.slider = this._viewQuery_daySlider_0.first;
    this.init((null as any),((<any>this.renderer).directRenderer? (null as any): [
      this._text_0,
      this._el_1,
      this._text_2,
      this._el_3,
      this._text_4,
      this._el_5,
      this._text_6,
      this._el_7,
      this._text_8,
      this._text_9,
      this._el_10,
      this._text_11,
      this._el_12,
      this._text_13,
      this._el_14,
      this._text_15,
      this._el_16,
      this._text_17,
      this._anchor_18,
      this._text_19,
      this._anchor_20,
      this._text_21,
      this._text_22,
      this._text_23,
      this._text_24,
      this._text_25,
      this._text_26,
      this._el_27,
      this._text_28,
      this._anchor_29,
      this._text_30,
      this._anchor_31,
      this._text_32,
      this._text_33,
      this._text_34,
      this._el_35,
      this._text_36,
      this._el_37,
      this._text_38,
      this._el_39,
      this._text_40,
      this._text_41,
      this._el_42,
      this._text_43,
      this._el_44,
      this._text_45,
      this._el_46,
      this._text_47,
      this._el_48,
      this._text_49,
      this._anchor_50,
      this._text_51,
      this._anchor_52,
      this._text_53,
      this._text_54,
      this._text_55,
      this._text_56,
      this._text_57,
      this._text_58,
      this._el_59,
      this._text_60,
      this._anchor_61,
      this._text_62,
      this._anchor_63,
      this._text_64,
      this._text_65,
      this._text_66,
      this._el_67,
      this._text_68,
      this._el_69,
      this._text_70,
      this._el_71,
      this._text_72,
      this._text_73,
      this._el_74,
      this._text_75,
      this._el_76,
      this._text_77,
      this._el_78,
      this._text_79,
      this._el_80,
      this._text_81,
      this._anchor_82,
      this._text_83,
      this._anchor_84,
      this._text_85,
      this._text_86,
      this._text_87,
      this._text_88,
      this._text_89,
      this._text_90,
      this._el_91,
      this._text_92,
      this._anchor_93,
      this._text_94,
      this._anchor_95,
      this._text_96,
      this._text_97,
      this._text_98,
      this._text_99
    ]
    ),[disposable_0]);
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import24.TemplateRef) && (18 === requestNodeIndex))) { return this._TemplateRef_18_5; }
    if (((token === import26.NgIf) && (18 === requestNodeIndex))) { return this._NgIf_18_6.context; }
    if (((token === import24.TemplateRef) && (20 === requestNodeIndex))) { return this._TemplateRef_20_5; }
    if (((token === import26.NgIf) && (20 === requestNodeIndex))) { return this._NgIf_20_6.context; }
    if (((token === import14.Scroll) && ((10 <= requestNodeIndex) && (requestNodeIndex <= 24)))) { return this._Scroll_10_3.context; }
    if (((token === import24.TemplateRef) && (29 === requestNodeIndex))) { return this._TemplateRef_29_5; }
    if (((token === import26.NgIf) && (29 === requestNodeIndex))) { return this._NgIf_29_6.context; }
    if (((token === import24.TemplateRef) && (31 === requestNodeIndex))) { return this._TemplateRef_31_5; }
    if (((token === import26.NgIf) && (31 === requestNodeIndex))) { return this._NgIf_31_6.context; }
    if (((token === import14.Scroll) && ((27 <= requestNodeIndex) && (requestNodeIndex <= 32)))) { return this._Scroll_27_3.context; }
    if (((token === import12.Slide) && ((3 <= requestNodeIndex) && (requestNodeIndex <= 33)))) { return this._Slide_3_3.context; }
    if (((token === import24.TemplateRef) && (50 === requestNodeIndex))) { return this._TemplateRef_50_5; }
    if (((token === import26.NgIf) && (50 === requestNodeIndex))) { return this._NgIf_50_6.context; }
    if (((token === import24.TemplateRef) && (52 === requestNodeIndex))) { return this._TemplateRef_52_5; }
    if (((token === import26.NgIf) && (52 === requestNodeIndex))) { return this._NgIf_52_6.context; }
    if (((token === import14.Scroll) && ((42 <= requestNodeIndex) && (requestNodeIndex <= 56)))) { return this._Scroll_42_3.context; }
    if (((token === import24.TemplateRef) && (61 === requestNodeIndex))) { return this._TemplateRef_61_5; }
    if (((token === import26.NgIf) && (61 === requestNodeIndex))) { return this._NgIf_61_6.context; }
    if (((token === import24.TemplateRef) && (63 === requestNodeIndex))) { return this._TemplateRef_63_5; }
    if (((token === import26.NgIf) && (63 === requestNodeIndex))) { return this._NgIf_63_6.context; }
    if (((token === import14.Scroll) && ((59 <= requestNodeIndex) && (requestNodeIndex <= 64)))) { return this._Scroll_59_3.context; }
    if (((token === import12.Slide) && ((35 <= requestNodeIndex) && (requestNodeIndex <= 65)))) { return this._Slide_35_3.context; }
    if (((token === import24.TemplateRef) && (82 === requestNodeIndex))) { return this._TemplateRef_82_5; }
    if (((token === import26.NgIf) && (82 === requestNodeIndex))) { return this._NgIf_82_6.context; }
    if (((token === import24.TemplateRef) && (84 === requestNodeIndex))) { return this._TemplateRef_84_5; }
    if (((token === import26.NgIf) && (84 === requestNodeIndex))) { return this._NgIf_84_6.context; }
    if (((token === import14.Scroll) && ((74 <= requestNodeIndex) && (requestNodeIndex <= 88)))) { return this._Scroll_74_3.context; }
    if (((token === import24.TemplateRef) && (93 === requestNodeIndex))) { return this._TemplateRef_93_5; }
    if (((token === import26.NgIf) && (93 === requestNodeIndex))) { return this._NgIf_93_6.context; }
    if (((token === import24.TemplateRef) && (95 === requestNodeIndex))) { return this._TemplateRef_95_5; }
    if (((token === import26.NgIf) && (95 === requestNodeIndex))) { return this._NgIf_95_6.context; }
    if (((token === import14.Scroll) && ((91 <= requestNodeIndex) && (requestNodeIndex <= 96)))) { return this._Scroll_91_3.context; }
    if (((token === import12.Slide) && ((67 <= requestNodeIndex) && (requestNodeIndex <= 97)))) { return this._Slide_67_3.context; }
    if (((token === import10.Slides) && ((1 <= requestNodeIndex) && (requestNodeIndex <= 98)))) { return this._Slides_1_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_1_0_0:any = true;
    this._Slides_1_3.check_loop(currVal_1_0_0,throwOnChange,false);
    if (this._Slides_1_3.ngDoCheck(this,this._el_1,throwOnChange)) { this.compView_1.markAsCheckOnce(); }
    if (this._Slide_3_3.ngDoCheck(this,this._el_3,throwOnChange)) { this.compView_3.markAsCheckOnce(); }
    const currVal_10_0_0:any = 'true';
    this._Scroll_10_3.check_scrollY(currVal_10_0_0,throwOnChange,false);
    const currVal_10_0_1:any = 'false';
    this._Scroll_10_3.check_zoom(currVal_10_0_1,throwOnChange,false);
    if (this._Scroll_10_3.ngDoCheck(this,this._el_10,throwOnChange)) { this.compView_10.markAsCheckOnce(); }
    const currVal_18_0_0:any = (0 === this.context.currentViewIndex);
    this._NgIf_18_6.check_ngIf(currVal_18_0_0,throwOnChange,false);
    this._NgIf_18_6.ngDoCheck(this,this._anchor_18,throwOnChange);
    const currVal_20_0_0:any = (0 !== this.context.currentViewIndex);
    this._NgIf_20_6.check_ngIf(currVal_20_0_0,throwOnChange,false);
    this._NgIf_20_6.ngDoCheck(this,this._anchor_20,throwOnChange);
    const currVal_27_0_0:any = 'true';
    this._Scroll_27_3.check_scrollY(currVal_27_0_0,throwOnChange,false);
    if (this._Scroll_27_3.ngDoCheck(this,this._el_27,throwOnChange)) { this.compView_27.markAsCheckOnce(); }
    const currVal_29_0_0:any = (0 === this.context.currentViewIndex);
    this._NgIf_29_6.check_ngIf(currVal_29_0_0,throwOnChange,false);
    this._NgIf_29_6.ngDoCheck(this,this._anchor_29,throwOnChange);
    const currVal_31_0_0:any = (0 !== this.context.currentViewIndex);
    this._NgIf_31_6.check_ngIf(currVal_31_0_0,throwOnChange,false);
    this._NgIf_31_6.ngDoCheck(this,this._anchor_31,throwOnChange);
    if (this._Slide_35_3.ngDoCheck(this,this._el_35,throwOnChange)) { this.compView_35.markAsCheckOnce(); }
    const currVal_42_0_0:any = 'true';
    this._Scroll_42_3.check_scrollY(currVal_42_0_0,throwOnChange,false);
    const currVal_42_0_1:any = 'false';
    this._Scroll_42_3.check_zoom(currVal_42_0_1,throwOnChange,false);
    if (this._Scroll_42_3.ngDoCheck(this,this._el_42,throwOnChange)) { this.compView_42.markAsCheckOnce(); }
    const currVal_50_0_0:any = (1 === this.context.currentViewIndex);
    this._NgIf_50_6.check_ngIf(currVal_50_0_0,throwOnChange,false);
    this._NgIf_50_6.ngDoCheck(this,this._anchor_50,throwOnChange);
    const currVal_52_0_0:any = (1 !== this.context.currentViewIndex);
    this._NgIf_52_6.check_ngIf(currVal_52_0_0,throwOnChange,false);
    this._NgIf_52_6.ngDoCheck(this,this._anchor_52,throwOnChange);
    const currVal_59_0_0:any = 'true';
    this._Scroll_59_3.check_scrollY(currVal_59_0_0,throwOnChange,false);
    if (this._Scroll_59_3.ngDoCheck(this,this._el_59,throwOnChange)) { this.compView_59.markAsCheckOnce(); }
    const currVal_61_0_0:any = (1 === this.context.currentViewIndex);
    this._NgIf_61_6.check_ngIf(currVal_61_0_0,throwOnChange,false);
    this._NgIf_61_6.ngDoCheck(this,this._anchor_61,throwOnChange);
    const currVal_63_0_0:any = (1 !== this.context.currentViewIndex);
    this._NgIf_63_6.check_ngIf(currVal_63_0_0,throwOnChange,false);
    this._NgIf_63_6.ngDoCheck(this,this._anchor_63,throwOnChange);
    if (this._Slide_67_3.ngDoCheck(this,this._el_67,throwOnChange)) { this.compView_67.markAsCheckOnce(); }
    const currVal_74_0_0:any = 'true';
    this._Scroll_74_3.check_scrollY(currVal_74_0_0,throwOnChange,false);
    const currVal_74_0_1:any = 'false';
    this._Scroll_74_3.check_zoom(currVal_74_0_1,throwOnChange,false);
    if (this._Scroll_74_3.ngDoCheck(this,this._el_74,throwOnChange)) { this.compView_74.markAsCheckOnce(); }
    const currVal_82_0_0:any = (2 === this.context.currentViewIndex);
    this._NgIf_82_6.check_ngIf(currVal_82_0_0,throwOnChange,false);
    this._NgIf_82_6.ngDoCheck(this,this._anchor_82,throwOnChange);
    const currVal_84_0_0:any = (2 !== this.context.currentViewIndex);
    this._NgIf_84_6.check_ngIf(currVal_84_0_0,throwOnChange,false);
    this._NgIf_84_6.ngDoCheck(this,this._anchor_84,throwOnChange);
    const currVal_91_0_0:any = 'true';
    this._Scroll_91_3.check_scrollY(currVal_91_0_0,throwOnChange,false);
    if (this._Scroll_91_3.ngDoCheck(this,this._el_91,throwOnChange)) { this.compView_91.markAsCheckOnce(); }
    const currVal_93_0_0:any = (2 === this.context.currentViewIndex);
    this._NgIf_93_6.check_ngIf(currVal_93_0_0,throwOnChange,false);
    this._NgIf_93_6.ngDoCheck(this,this._anchor_93,throwOnChange);
    const currVal_95_0_0:any = (2 !== this.context.currentViewIndex);
    this._NgIf_95_6.check_ngIf(currVal_95_0_0,throwOnChange,false);
    this._NgIf_95_6.ngDoCheck(this,this._anchor_95,throwOnChange);
    this._vc_18.detectChangesInNestedViews(throwOnChange);
    this._vc_20.detectChangesInNestedViews(throwOnChange);
    this._vc_29.detectChangesInNestedViews(throwOnChange);
    this._vc_31.detectChangesInNestedViews(throwOnChange);
    this._vc_50.detectChangesInNestedViews(throwOnChange);
    this._vc_52.detectChangesInNestedViews(throwOnChange);
    this._vc_61.detectChangesInNestedViews(throwOnChange);
    this._vc_63.detectChangesInNestedViews(throwOnChange);
    this._vc_82.detectChangesInNestedViews(throwOnChange);
    this._vc_84.detectChangesInNestedViews(throwOnChange);
    this._vc_93.detectChangesInNestedViews(throwOnChange);
    this._vc_95.detectChangesInNestedViews(throwOnChange);
    if (!throwOnChange) { if ((this.numberOfChecks === 0)) { this._Slides_1_3.context.ngAfterContentInit(); } }
    const currVal_157:any = import3.inlineInterpolate(1,'',this.context.allDayLabel,'');
    if (import3.checkBinding(throwOnChange,this._expr_157,currVal_157)) {
      this.renderer.setText(this._text_8,currVal_157);
      this._expr_157 = currVal_157;
    }
    this._Scroll_10_3.checkHost(this,this.compView_10,this._el_10,throwOnChange);
    this._Scroll_27_3.checkHost(this,this.compView_27,this._el_27,throwOnChange);
    const currVal_159:any = import3.inlineInterpolate(1,'',this.context.allDayLabel,'');
    if (import3.checkBinding(throwOnChange,this._expr_159,currVal_159)) {
      this.renderer.setText(this._text_40,currVal_159);
      this._expr_159 = currVal_159;
    }
    this._Scroll_42_3.checkHost(this,this.compView_42,this._el_42,throwOnChange);
    this._Scroll_59_3.checkHost(this,this.compView_59,this._el_59,throwOnChange);
    const currVal_160:any = import3.inlineInterpolate(1,'',this.context.allDayLabel,'');
    if (import3.checkBinding(throwOnChange,this._expr_160,currVal_160)) {
      this.renderer.setText(this._text_72,currVal_160);
      this._expr_160 = currVal_160;
    }
    this._Scroll_74_3.checkHost(this,this.compView_74,this._el_74,throwOnChange);
    this._Scroll_91_3.checkHost(this,this.compView_91,this._el_91,throwOnChange);
    this.compView_1.detectChanges(throwOnChange);
    this.compView_3.detectChanges(throwOnChange);
    this.compView_10.detectChanges(throwOnChange);
    this.compView_27.detectChanges(throwOnChange);
    this.compView_35.detectChanges(throwOnChange);
    this.compView_42.detectChanges(throwOnChange);
    this.compView_59.detectChanges(throwOnChange);
    this.compView_67.detectChanges(throwOnChange);
    this.compView_74.detectChanges(throwOnChange);
    this.compView_91.detectChanges(throwOnChange);
  }
  destroyInternal():void {
    this._vc_18.destroyNestedViews();
    this._vc_20.destroyNestedViews();
    this._vc_29.destroyNestedViews();
    this._vc_31.destroyNestedViews();
    this._vc_50.destroyNestedViews();
    this._vc_52.destroyNestedViews();
    this._vc_61.destroyNestedViews();
    this._vc_63.destroyNestedViews();
    this._vc_82.destroyNestedViews();
    this._vc_84.destroyNestedViews();
    this._vc_93.destroyNestedViews();
    this._vc_95.destroyNestedViews();
    this.compView_1.destroy();
    this.compView_3.destroy();
    this.compView_10.destroy();
    this.compView_27.destroy();
    this.compView_35.destroy();
    this.compView_42.destroy();
    this.compView_59.destroy();
    this.compView_67.destroy();
    this.compView_74.destroy();
    this.compView_91.destroy();
    this._Slide_3_3.ngOnDestroy();
    this._Slide_35_3.ngOnDestroy();
    this._Slide_67_3.ngOnDestroy();
    this._Slides_1_3.ngOnDestroy();
  }
  visitProjectableNodesInternal(nodeIndex:number,ngContentIndex:number,cb:any,ctx:any):void {
    if (((nodeIndex == 1) && (ngContentIndex == 0))) {
      cb(this._text_2,ctx);
      cb(this._el_3,ctx);
      cb(this._text_34,ctx);
      cb(this._el_35,ctx);
      cb(this._text_66,ctx);
      cb(this._el_67,ctx);
      cb(this._text_98,ctx);
    }
    if (((nodeIndex == 3) && (ngContentIndex == 0))) {
      cb(this._text_4,ctx);
      cb(this._el_5,ctx);
      cb(this._text_26,ctx);
      cb(this._el_27,ctx);
      cb(this._text_33,ctx);
    }
    if (((nodeIndex == 10) && (ngContentIndex == 0))) {
      cb(this._text_11,ctx);
      cb(this._el_12,ctx);
      cb(this._text_24,ctx);
    }
    if (((nodeIndex == 27) && (ngContentIndex == 0))) {
      cb(this._text_28,ctx);
      cb(this._vc_29.nativeElement,ctx);
      this._vc_29.visitNestedViewRootNodes(cb,ctx);
      cb(this._text_30,ctx);
      cb(this._vc_31.nativeElement,ctx);
      this._vc_31.visitNestedViewRootNodes(cb,ctx);
      cb(this._text_32,ctx);
    }
    if (((nodeIndex == 35) && (ngContentIndex == 0))) {
      cb(this._text_36,ctx);
      cb(this._el_37,ctx);
      cb(this._text_58,ctx);
      cb(this._el_59,ctx);
      cb(this._text_65,ctx);
    }
    if (((nodeIndex == 42) && (ngContentIndex == 0))) {
      cb(this._text_43,ctx);
      cb(this._el_44,ctx);
      cb(this._text_56,ctx);
    }
    if (((nodeIndex == 59) && (ngContentIndex == 0))) {
      cb(this._text_60,ctx);
      cb(this._vc_61.nativeElement,ctx);
      this._vc_61.visitNestedViewRootNodes(cb,ctx);
      cb(this._text_62,ctx);
      cb(this._vc_63.nativeElement,ctx);
      this._vc_63.visitNestedViewRootNodes(cb,ctx);
      cb(this._text_64,ctx);
    }
    if (((nodeIndex == 67) && (ngContentIndex == 0))) {
      cb(this._text_68,ctx);
      cb(this._el_69,ctx);
      cb(this._text_90,ctx);
      cb(this._el_91,ctx);
      cb(this._text_97,ctx);
    }
    if (((nodeIndex == 74) && (ngContentIndex == 0))) {
      cb(this._text_75,ctx);
      cb(this._el_76,ctx);
      cb(this._text_88,ctx);
    }
    if (((nodeIndex == 91) && (ngContentIndex == 0))) {
      cb(this._text_92,ctx);
      cb(this._vc_93.nativeElement,ctx);
      this._vc_93.visitNestedViewRootNodes(cb,ctx);
      cb(this._text_94,ctx);
      cb(this._vc_95.nativeElement,ctx);
      this._vc_95.visitNestedViewRootNodes(cb,ctx);
      cb(this._text_96,ctx);
    }
  }
  createEmbeddedViewInternal(nodeIndex:number):import2.AppView<any> {
    if ((nodeIndex == 18)) { return new View_DayViewComponent1(this.viewUtils,this,18,this._anchor_18,this._vc_18); }
    if ((nodeIndex == 20)) { return new View_DayViewComponent4(this.viewUtils,this,20,this._anchor_20,this._vc_20); }
    if ((nodeIndex == 29)) { return new View_DayViewComponent5(this.viewUtils,this,29,this._anchor_29,this._vc_29); }
    if ((nodeIndex == 31)) { return new View_DayViewComponent10(this.viewUtils,this,31,this._anchor_31,this._vc_31); }
    if ((nodeIndex == 50)) { return new View_DayViewComponent12(this.viewUtils,this,50,this._anchor_50,this._vc_50); }
    if ((nodeIndex == 52)) { return new View_DayViewComponent15(this.viewUtils,this,52,this._anchor_52,this._vc_52); }
    if ((nodeIndex == 61)) { return new View_DayViewComponent16(this.viewUtils,this,61,this._anchor_61,this._vc_61); }
    if ((nodeIndex == 63)) { return new View_DayViewComponent21(this.viewUtils,this,63,this._anchor_63,this._vc_63); }
    if ((nodeIndex == 82)) { return new View_DayViewComponent23(this.viewUtils,this,82,this._anchor_82,this._vc_82); }
    if ((nodeIndex == 84)) { return new View_DayViewComponent26(this.viewUtils,this,84,this._anchor_84,this._vc_84); }
    if ((nodeIndex == 93)) { return new View_DayViewComponent27(this.viewUtils,this,93,this._anchor_93,this._vc_93); }
    if ((nodeIndex == 95)) { return new View_DayViewComponent32(this.viewUtils,this,95,this._anchor_95,this._vc_95); }
    return (null as any);
  }
  handleEvent_1(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'ionSlideDidChange')) {
      const pd_sub_0:any = ((<any>this.context.onSlideChanged()) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
}
class View_DayViewComponent1 extends import2.AppView<any> {
  _el_0:any;
  _NgClass_0_3:import27.Wrapper_NgClass;
  _NgStyle_0_4:import28.Wrapper_NgStyle;
  _text_1:any;
  _anchor_2:any;
  /*private*/ _vc_2:import16.ViewContainer;
  _TemplateRef_2_5:any;
  _NgFor_2_6:import29.Wrapper_NgFor;
  _text_3:any;
  _map_9:any;
  _map_10:any;
  constructor(viewUtils:import3.ViewUtils,parentView:import2.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import16.ViewContainer) {
    super(View_DayViewComponent1,renderType_DayViewComponent,import6.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import1.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._map_9 = import3.pureProxy1((p0:any):{[key: string]:any} => {
      return {'calendar-event-wrap': p0};
    });
    this._map_10 = import3.pureProxy1((p0:any):{[key: string]:any} => {
      return {height: p0};
    });
  }
  createInternal(rootSelector:string):import7.ComponentRef<any> {
    this._el_0 = import3.createRenderElement(this.renderer,(null as any),'td',new import3.InlineArray2(2,'class','calendar-cell'),(null as any));
    this._NgClass_0_3 = new import27.Wrapper_NgClass(this.parentView.parentView.injectorGet(import30.IterableDiffers,this.parentView.parentIndex),this.parentView.parentView.injectorGet(import31.KeyValueDiffers,this.parentView.parentIndex),new import23.ElementRef(this._el_0),this.renderer);
    this._NgStyle_0_4 = new import28.Wrapper_NgStyle(this.parentView.parentView.injectorGet(import31.KeyValueDiffers,this.parentView.parentIndex),new import23.ElementRef(this._el_0),this.renderer);
    this._text_1 = this.renderer.createText(this._el_0,'\n                                    ',(null as any));
    this._anchor_2 = this.renderer.createTemplateAnchor(this._el_0,(null as any));
    this._vc_2 = new import16.ViewContainer(2,0,this,this._anchor_2);
    this._TemplateRef_2_5 = new import24.TemplateRef_(this,2,this._anchor_2);
    this._NgFor_2_6 = new import29.Wrapper_NgFor(this._vc_2.vcRef,this._TemplateRef_2_5,this.parentView.parentView.injectorGet(import30.IterableDiffers,this.parentView.parentIndex),this.parentView.ref);
    this._text_3 = this.renderer.createText(this._el_0,'\n                                ',(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._anchor_2,
      this._text_3
    ]
    ),(null as any));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import24.TemplateRef) && (2 === requestNodeIndex))) { return this._TemplateRef_2_5; }
    if (((token === import32.NgFor) && (2 === requestNodeIndex))) { return this._NgFor_2_6.context; }
    if (((token === import33.NgClass) && ((0 <= requestNodeIndex) && (requestNodeIndex <= 3)))) { return this._NgClass_0_3.context; }
    if (((token === import34.NgStyle) && ((0 <= requestNodeIndex) && (requestNodeIndex <= 3)))) { return this._NgStyle_0_4.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_0_0_0:any = 'calendar-cell';
    this._NgClass_0_3.check_klass(currVal_0_0_0,throwOnChange,false);
    const currVal_0_0_1:any = this._map_9((this.parentView.context.views[0].allDayEvents.length > 0));
    this._NgClass_0_3.check_ngClass(currVal_0_0_1,throwOnChange,false);
    this._NgClass_0_3.ngDoCheck(this,this._el_0,throwOnChange);
    const currVal_0_1_0:any = this._map_10(((25 * this.parentView.context.views[0].allDayEvents.length) + 'px'));
    this._NgStyle_0_4.check_ngStyle(currVal_0_1_0,throwOnChange,false);
    this._NgStyle_0_4.ngDoCheck(this,this._el_0,throwOnChange);
    const currVal_2_0_0:any = this.parentView.context.views[0].allDayEvents;
    this._NgFor_2_6.check_ngForOf(currVal_2_0_0,throwOnChange,false);
    this._NgFor_2_6.ngDoCheck(this,this._anchor_2,throwOnChange);
    this._vc_2.detectChangesInNestedViews(throwOnChange);
  }
  destroyInternal():void {
    this._vc_2.destroyNestedViews();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  createEmbeddedViewInternal(nodeIndex:number):import2.AppView<any> {
    if ((nodeIndex == 2)) { return new View_DayViewComponent2(this.viewUtils,this,2,this._anchor_2,this._vc_2); }
    return (null as any);
  }
}
class View_DayViewComponent2 extends import2.AppView<any> {
  _el_0:any;
  _NgStyle_0_3:import28.Wrapper_NgStyle;
  _text_1:any;
  _anchor_2:any;
  /*private*/ _vc_2:import16.ViewContainer;
  _TemplateRef_2_5:any;
  _NgTemplateOutlet_2_6:import35.Wrapper_NgTemplateOutlet;
  _text_3:any;
  _map_8:any;
  _map_9:any;
  constructor(viewUtils:import3.ViewUtils,parentView:import2.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import16.ViewContainer) {
    super(View_DayViewComponent2,renderType_DayViewComponent,import6.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import1.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._map_8 = import3.pureProxy3((p0:any,p1:any,p2:any):{[key: string]:any} => {
      return {
        top: p0,
        width: p1,
        height: p2
      }
      ;
    });
    this._map_9 = import3.pureProxy1((p0:any):{[key: string]:any} => {
      return {displayEvent: p0};
    });
  }
  createInternal(rootSelector:string):import7.ComponentRef<any> {
    this._el_0 = import3.createRenderElement(this.renderer,(null as any),'div',new import3.InlineArray4(4,'class','calendar-event','tappable',''),(null as any));
    this._NgStyle_0_3 = new import28.Wrapper_NgStyle(this.parentView.parentView.parentView.injectorGet(import31.KeyValueDiffers,this.parentView.parentView.parentIndex),new import23.ElementRef(this._el_0),this.renderer);
    this._text_1 = this.renderer.createText(this._el_0,'\n                                        ',(null as any));
    this._anchor_2 = this.renderer.createTemplateAnchor(this._el_0,(null as any));
    this._vc_2 = new import16.ViewContainer(2,0,this,this._anchor_2);
    this._TemplateRef_2_5 = new import24.TemplateRef_(this,2,this._anchor_2);
    this._NgTemplateOutlet_2_6 = new import35.Wrapper_NgTemplateOutlet(this._vc_2.vcRef);
    this._text_3 = this.renderer.createText(this._el_0,'\n                                    ',(null as any));
    var disposable_0:Function = import3.subscribeToRenderElement(this,this._el_0,new import3.InlineArray2(2,'click',(null as any)),this.eventHandler(this.handleEvent_0));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._anchor_2,
      this._text_3
    ]
    ),[disposable_0]);
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import24.TemplateRef) && (2 === requestNodeIndex))) { return this._TemplateRef_2_5; }
    if (((token === import36.NgTemplateOutlet) && (2 === requestNodeIndex))) { return this._NgTemplateOutlet_2_6.context; }
    if (((token === import34.NgStyle) && ((0 <= requestNodeIndex) && (requestNodeIndex <= 3)))) { return this._NgStyle_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_0_0_0:any = this._map_8(((25 * this.context.index) + 'px'),'100%','25px');
    this._NgStyle_0_3.check_ngStyle(currVal_0_0_0,throwOnChange,false);
    this._NgStyle_0_3.ngDoCheck(this,this._el_0,throwOnChange);
    const currVal_2_0_0:any = this._map_9(this.context.$implicit);
    this._NgTemplateOutlet_2_6.check_ngOutletContext(currVal_2_0_0,throwOnChange,false);
    const currVal_2_0_1:any = this.parentView.parentView.context.dayviewAllDayEventTemplate;
    this._NgTemplateOutlet_2_6.check_ngTemplateOutlet(currVal_2_0_1,throwOnChange,false);
    this._NgTemplateOutlet_2_6.ngDoCheck(this,this._anchor_2,throwOnChange);
    this._vc_2.detectChangesInNestedViews(throwOnChange);
  }
  destroyInternal():void {
    this._vc_2.destroyNestedViews();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  createEmbeddedViewInternal(nodeIndex:number):import2.AppView<any> {
    if ((nodeIndex == 2)) { return new View_DayViewComponent3(this.viewUtils,this,2,this._anchor_2,this._vc_2); }
    return (null as any);
  }
  handleEvent_0(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.parentView.parentView.context.eventSelected(this.context.$implicit.event)) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
}
class View_DayViewComponent3 extends import2.AppView<any> {
  _text_0:any;
  constructor(viewUtils:import3.ViewUtils,parentView:import2.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import16.ViewContainer) {
    super(View_DayViewComponent3,renderType_DayViewComponent,import6.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import1.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
  }
  createInternal(rootSelector:string):import7.ComponentRef<any> {
    this._text_0 = this.renderer.createText((null as any),'\n                                        ',(null as any));
    this.init(this._text_0,((<any>this.renderer).directRenderer? (null as any): [this._text_0]),(null as any));
    return (null as any);
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._text_0,ctx);
  }
}
class View_DayViewComponent4 extends import2.AppView<any> {
  _el_0:any;
  _text_1:any;
  constructor(viewUtils:import3.ViewUtils,parentView:import2.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import16.ViewContainer) {
    super(View_DayViewComponent4,renderType_DayViewComponent,import6.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import1.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
  }
  createInternal(rootSelector:string):import7.ComponentRef<any> {
    this._el_0 = import3.createRenderElement(this.renderer,(null as any),'td',new import3.InlineArray2(2,'class','calendar-cell'),(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'\n                                ',(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1
    ]
    ),(null as any));
    return (null as any);
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
class View_DayViewComponent5 extends import2.AppView<any> {
  _el_0:any;
  _text_1:any;
  _el_2:any;
  _text_3:any;
  _anchor_4:any;
  /*private*/ _vc_4:import16.ViewContainer;
  _TemplateRef_4_5:any;
  _NgFor_4_6:import29.Wrapper_NgFor;
  _text_5:any;
  _text_6:any;
  constructor(viewUtils:import3.ViewUtils,parentView:import2.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import16.ViewContainer) {
    super(View_DayViewComponent5,renderType_DayViewComponent,import6.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import1.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
  }
  createInternal(rootSelector:string):import7.ComponentRef<any> {
    this._el_0 = import3.createRenderElement(this.renderer,(null as any),'table',new import3.InlineArray2(2,'class','table table-bordered table-fixed dayview-normal-event-table'),(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'\n                        ',(null as any));
    this._el_2 = import3.createRenderElement(this.renderer,this._el_0,'tbody',import3.EMPTY_INLINE_ARRAY,(null as any));
    this._text_3 = this.renderer.createText(this._el_2,'\n                        ',(null as any));
    this._anchor_4 = this.renderer.createTemplateAnchor(this._el_2,(null as any));
    this._vc_4 = new import16.ViewContainer(4,2,this,this._anchor_4);
    this._TemplateRef_4_5 = new import24.TemplateRef_(this,4,this._anchor_4);
    this._NgFor_4_6 = new import29.Wrapper_NgFor(this._vc_4.vcRef,this._TemplateRef_4_5,this.parentView.parentView.injectorGet(import30.IterableDiffers,this.parentView.parentIndex),this.parentView.ref);
    this._text_5 = this.renderer.createText(this._el_2,'\n                        ',(null as any));
    this._text_6 = this.renderer.createText(this._el_0,'\n                    ',(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._el_2,
      this._text_3,
      this._anchor_4,
      this._text_5,
      this._text_6
    ]
    ),(null as any));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import24.TemplateRef) && (4 === requestNodeIndex))) { return this._TemplateRef_4_5; }
    if (((token === import32.NgFor) && (4 === requestNodeIndex))) { return this._NgFor_4_6.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_4_0_0:any = this.parentView.context.views[0].rows;
    this._NgFor_4_6.check_ngForOf(currVal_4_0_0,throwOnChange,false);
    this._NgFor_4_6.ngDoCheck(this,this._anchor_4,throwOnChange);
    this._vc_4.detectChangesInNestedViews(throwOnChange);
  }
  destroyInternal():void {
    this._vc_4.destroyNestedViews();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  createEmbeddedViewInternal(nodeIndex:number):import2.AppView<any> {
    if ((nodeIndex == 4)) { return new View_DayViewComponent6(this.viewUtils,this,4,this._anchor_4,this._vc_4); }
    return (null as any);
  }
}
class View_DayViewComponent6 extends import2.AppView<any> {
  _el_0:any;
  _text_1:any;
  _el_2:any;
  _text_3:any;
  _text_4:any;
  _el_5:any;
  _text_6:any;
  _anchor_7:any;
  /*private*/ _vc_7:import16.ViewContainer;
  _TemplateRef_7_5:any;
  _NgIf_7_6:import17.Wrapper_NgIf;
  _text_8:any;
  _text_9:any;
  /*private*/ _expr_13:any;
  _pipe_date_0_0:any;
  constructor(viewUtils:import3.ViewUtils,parentView:import2.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import16.ViewContainer) {
    super(View_DayViewComponent6,renderType_DayViewComponent,import6.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import1.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._expr_13 = import1.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import7.ComponentRef<any> {
    this._el_0 = import3.createRenderElement(this.renderer,(null as any),'tr',import3.EMPTY_INLINE_ARRAY,(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'\n                            ',(null as any));
    this._el_2 = import3.createRenderElement(this.renderer,this._el_0,'td',new import3.InlineArray2(2,'class','calendar-hour-column text-center'),(null as any));
    this._text_3 = this.renderer.createText(this._el_2,'',(null as any));
    this._text_4 = this.renderer.createText(this._el_0,'\n                            ',(null as any));
    this._el_5 = import3.createRenderElement(this.renderer,this._el_0,'td',new import3.InlineArray4(4,'class','calendar-cell','tappable',''),(null as any));
    this._text_6 = this.renderer.createText(this._el_5,'\n                                ',(null as any));
    this._anchor_7 = this.renderer.createTemplateAnchor(this._el_5,(null as any));
    this._vc_7 = new import16.ViewContainer(7,5,this,this._anchor_7);
    this._TemplateRef_7_5 = new import24.TemplateRef_(this,7,this._anchor_7);
    this._NgIf_7_6 = new import17.Wrapper_NgIf(this._vc_7.vcRef,this._TemplateRef_7_5);
    this._text_8 = this.renderer.createText(this._el_5,'\n                            ',(null as any));
    this._text_9 = this.renderer.createText(this._el_0,'\n                        ',(null as any));
    this._pipe_date_0_0 = import3.pureProxy2((<View_DayViewComponent0>this.parentView.parentView)._pipe_date_0.transform.bind((<View_DayViewComponent0>this.parentView.parentView)._pipe_date_0));
    var disposable_0:Function = import3.subscribeToRenderElement(this,this._el_5,new import3.InlineArray2(2,'click',(null as any)),this.eventHandler(this.handleEvent_5));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._el_2,
      this._text_3,
      this._text_4,
      this._el_5,
      this._text_6,
      this._anchor_7,
      this._text_8,
      this._text_9
    ]
    ),[disposable_0]);
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import24.TemplateRef) && (7 === requestNodeIndex))) { return this._TemplateRef_7_5; }
    if (((token === import26.NgIf) && (7 === requestNodeIndex))) { return this._NgIf_7_6.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const valUnwrapper:any = new import1.ValueUnwrapper();
    const currVal_7_0_0:any = this.context.$implicit.events;
    this._NgIf_7_6.check_ngIf(currVal_7_0_0,throwOnChange,false);
    this._NgIf_7_6.ngDoCheck(this,this._anchor_7,throwOnChange);
    this._vc_7.detectChangesInNestedViews(throwOnChange);
    valUnwrapper.reset();
    const currVal_13:any = import3.inlineInterpolate(1,'\n                                ',valUnwrapper.unwrap(import3.castByValue(this._pipe_date_0_0,(<View_DayViewComponent0>this.parentView.parentView)._pipe_date_0.transform)(this.context.$implicit.time,this.parentView.parentView.context.formatHourColumn)),'\n                            ');
    if ((valUnwrapper.hasWrappedValue || import3.checkBinding(throwOnChange,this._expr_13,currVal_13))) {
      this.renderer.setText(this._text_3,currVal_13);
      this._expr_13 = currVal_13;
    }
  }
  destroyInternal():void {
    this._vc_7.destroyNestedViews();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  createEmbeddedViewInternal(nodeIndex:number):import2.AppView<any> {
    if ((nodeIndex == 7)) { return new View_DayViewComponent7(this.viewUtils,this,7,this._anchor_7,this._vc_7); }
    return (null as any);
  }
  handleEvent_5(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.parentView.parentView.context.select(this.context.$implicit.time,this.context.$implicit.events)) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
}
class View_DayViewComponent7 extends import2.AppView<any> {
  _el_0:any;
  _NgClass_0_3:import27.Wrapper_NgClass;
  _text_1:any;
  _anchor_2:any;
  /*private*/ _vc_2:import16.ViewContainer;
  _TemplateRef_2_5:any;
  _NgFor_2_6:import29.Wrapper_NgFor;
  _text_3:any;
  _map_8:any;
  constructor(viewUtils:import3.ViewUtils,parentView:import2.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import16.ViewContainer) {
    super(View_DayViewComponent7,renderType_DayViewComponent,import6.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import1.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._map_8 = import3.pureProxy1((p0:any):{[key: string]:any} => {
      return {'calendar-event-wrap': p0};
    });
  }
  createInternal(rootSelector:string):import7.ComponentRef<any> {
    this._el_0 = import3.createRenderElement(this.renderer,(null as any),'div',import3.EMPTY_INLINE_ARRAY,(null as any));
    this._NgClass_0_3 = new import27.Wrapper_NgClass(this.parentView.parentView.parentView.parentView.injectorGet(import30.IterableDiffers,this.parentView.parentView.parentView.parentIndex),this.parentView.parentView.parentView.parentView.injectorGet(import31.KeyValueDiffers,this.parentView.parentView.parentView.parentIndex),new import23.ElementRef(this._el_0),this.renderer);
    this._text_1 = this.renderer.createText(this._el_0,'\n                                    ',(null as any));
    this._anchor_2 = this.renderer.createTemplateAnchor(this._el_0,(null as any));
    this._vc_2 = new import16.ViewContainer(2,0,this,this._anchor_2);
    this._TemplateRef_2_5 = new import24.TemplateRef_(this,2,this._anchor_2);
    this._NgFor_2_6 = new import29.Wrapper_NgFor(this._vc_2.vcRef,this._TemplateRef_2_5,this.parentView.parentView.parentView.parentView.injectorGet(import30.IterableDiffers,this.parentView.parentView.parentView.parentIndex),this.parentView.parentView.parentView.ref);
    this._text_3 = this.renderer.createText(this._el_0,'\n                                ',(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._anchor_2,
      this._text_3
    ]
    ),(null as any));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import24.TemplateRef) && (2 === requestNodeIndex))) { return this._TemplateRef_2_5; }
    if (((token === import32.NgFor) && (2 === requestNodeIndex))) { return this._NgFor_2_6.context; }
    if (((token === import33.NgClass) && ((0 <= requestNodeIndex) && (requestNodeIndex <= 3)))) { return this._NgClass_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_0_0_0:any = this._map_8(this.parentView.context.$implicit.events);
    this._NgClass_0_3.check_ngClass(currVal_0_0_0,throwOnChange,false);
    this._NgClass_0_3.ngDoCheck(this,this._el_0,throwOnChange);
    const currVal_2_0_0:any = this.parentView.context.$implicit.events;
    this._NgFor_2_6.check_ngForOf(currVal_2_0_0,throwOnChange,false);
    this._NgFor_2_6.ngDoCheck(this,this._anchor_2,throwOnChange);
    this._vc_2.detectChangesInNestedViews(throwOnChange);
  }
  destroyInternal():void {
    this._vc_2.destroyNestedViews();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  createEmbeddedViewInternal(nodeIndex:number):import2.AppView<any> {
    if ((nodeIndex == 2)) { return new View_DayViewComponent8(this.viewUtils,this,2,this._anchor_2,this._vc_2); }
    return (null as any);
  }
}
class View_DayViewComponent8 extends import2.AppView<any> {
  _el_0:any;
  _NgStyle_0_3:import28.Wrapper_NgStyle;
  _text_1:any;
  _anchor_2:any;
  /*private*/ _vc_2:import16.ViewContainer;
  _TemplateRef_2_5:any;
  _NgTemplateOutlet_2_6:import35.Wrapper_NgTemplateOutlet;
  _text_3:any;
  _map_8:any;
  _map_9:any;
  constructor(viewUtils:import3.ViewUtils,parentView:import2.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import16.ViewContainer) {
    super(View_DayViewComponent8,renderType_DayViewComponent,import6.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import1.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._map_8 = import3.pureProxy4((p0:any,p1:any,p2:any,p3:any):{[key: string]:any} => {
      return {
        top: p0,
        left: p1,
        width: p2,
        height: p3
      }
      ;
    });
    this._map_9 = import3.pureProxy1((p0:any):{[key: string]:any} => {
      return {displayEvent: p0};
    });
  }
  createInternal(rootSelector:string):import7.ComponentRef<any> {
    this._el_0 = import3.createRenderElement(this.renderer,(null as any),'div',new import3.InlineArray4(4,'class','calendar-event','tappable',''),(null as any));
    this._NgStyle_0_3 = new import28.Wrapper_NgStyle(this.parentView.parentView.parentView.parentView.parentView.injectorGet(import31.KeyValueDiffers,this.parentView.parentView.parentView.parentView.parentIndex),new import23.ElementRef(this._el_0),this.renderer);
    this._text_1 = this.renderer.createText(this._el_0,'\n                                         ',(null as any));
    this._anchor_2 = this.renderer.createTemplateAnchor(this._el_0,(null as any));
    this._vc_2 = new import16.ViewContainer(2,0,this,this._anchor_2);
    this._TemplateRef_2_5 = new import24.TemplateRef_(this,2,this._anchor_2);
    this._NgTemplateOutlet_2_6 = new import35.Wrapper_NgTemplateOutlet(this._vc_2.vcRef);
    this._text_3 = this.renderer.createText(this._el_0,'\n                                    ',(null as any));
    var disposable_0:Function = import3.subscribeToRenderElement(this,this._el_0,new import3.InlineArray2(2,'click',(null as any)),this.eventHandler(this.handleEvent_0));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._anchor_2,
      this._text_3
    ]
    ),[disposable_0]);
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import24.TemplateRef) && (2 === requestNodeIndex))) { return this._TemplateRef_2_5; }
    if (((token === import36.NgTemplateOutlet) && (2 === requestNodeIndex))) { return this._NgTemplateOutlet_2_6.context; }
    if (((token === import34.NgStyle) && ((0 <= requestNodeIndex) && (requestNodeIndex <= 3)))) { return this._NgStyle_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_0_0_0:any = this._map_8((((37 * this.context.$implicit.startOffset) / this.parentView.parentView.parentView.parentView.context.hourParts) + 'px'),(((100 / this.context.$implicit.overlapNumber) * this.context.$implicit.position) + '%'),((100 / this.context.$implicit.overlapNumber) + '%'),((37 * ((this.context.$implicit.endIndex - this.context.$implicit.startIndex) - ((this.context.$implicit.endOffset + this.context.$implicit.startOffset) / this.parentView.parentView.parentView.parentView.context.hourParts))) + 'px'));
    this._NgStyle_0_3.check_ngStyle(currVal_0_0_0,throwOnChange,false);
    this._NgStyle_0_3.ngDoCheck(this,this._el_0,throwOnChange);
    const currVal_2_0_0:any = this._map_9(this.context.$implicit);
    this._NgTemplateOutlet_2_6.check_ngOutletContext(currVal_2_0_0,throwOnChange,false);
    const currVal_2_0_1:any = this.parentView.parentView.parentView.parentView.context.dayviewNormalEventTemplate;
    this._NgTemplateOutlet_2_6.check_ngTemplateOutlet(currVal_2_0_1,throwOnChange,false);
    this._NgTemplateOutlet_2_6.ngDoCheck(this,this._anchor_2,throwOnChange);
    this._vc_2.detectChangesInNestedViews(throwOnChange);
  }
  destroyInternal():void {
    this._vc_2.destroyNestedViews();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  createEmbeddedViewInternal(nodeIndex:number):import2.AppView<any> {
    if ((nodeIndex == 2)) { return new View_DayViewComponent9(this.viewUtils,this,2,this._anchor_2,this._vc_2); }
    return (null as any);
  }
  handleEvent_0(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.parentView.parentView.parentView.parentView.context.eventSelected(this.context.$implicit.event)) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
}
class View_DayViewComponent9 extends import2.AppView<any> {
  _text_0:any;
  constructor(viewUtils:import3.ViewUtils,parentView:import2.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import16.ViewContainer) {
    super(View_DayViewComponent9,renderType_DayViewComponent,import6.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import1.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
  }
  createInternal(rootSelector:string):import7.ComponentRef<any> {
    this._text_0 = this.renderer.createText((null as any),'\n                                         ',(null as any));
    this.init(this._text_0,((<any>this.renderer).directRenderer? (null as any): [this._text_0]),(null as any));
    return (null as any);
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._text_0,ctx);
  }
}
class View_DayViewComponent10 extends import2.AppView<any> {
  _el_0:any;
  _text_1:any;
  _el_2:any;
  _text_3:any;
  _anchor_4:any;
  /*private*/ _vc_4:import16.ViewContainer;
  _TemplateRef_4_5:any;
  _NgFor_4_6:import29.Wrapper_NgFor;
  _text_5:any;
  _text_6:any;
  constructor(viewUtils:import3.ViewUtils,parentView:import2.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import16.ViewContainer) {
    super(View_DayViewComponent10,renderType_DayViewComponent,import6.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import1.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
  }
  createInternal(rootSelector:string):import7.ComponentRef<any> {
    this._el_0 = import3.createRenderElement(this.renderer,(null as any),'table',new import3.InlineArray2(2,'class','table table-bordered table-fixed dayview-normal-event-table'),(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'\n                        ',(null as any));
    this._el_2 = import3.createRenderElement(this.renderer,this._el_0,'tbody',import3.EMPTY_INLINE_ARRAY,(null as any));
    this._text_3 = this.renderer.createText(this._el_2,'\n                        ',(null as any));
    this._anchor_4 = this.renderer.createTemplateAnchor(this._el_2,(null as any));
    this._vc_4 = new import16.ViewContainer(4,2,this,this._anchor_4);
    this._TemplateRef_4_5 = new import24.TemplateRef_(this,4,this._anchor_4);
    this._NgFor_4_6 = new import29.Wrapper_NgFor(this._vc_4.vcRef,this._TemplateRef_4_5,this.parentView.parentView.injectorGet(import30.IterableDiffers,this.parentView.parentIndex),this.parentView.ref);
    this._text_5 = this.renderer.createText(this._el_2,'\n                        ',(null as any));
    this._text_6 = this.renderer.createText(this._el_0,'\n                    ',(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._el_2,
      this._text_3,
      this._anchor_4,
      this._text_5,
      this._text_6
    ]
    ),(null as any));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import24.TemplateRef) && (4 === requestNodeIndex))) { return this._TemplateRef_4_5; }
    if (((token === import32.NgFor) && (4 === requestNodeIndex))) { return this._NgFor_4_6.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_4_0_0:any = this.parentView.context.views[0].rows;
    this._NgFor_4_6.check_ngForOf(currVal_4_0_0,throwOnChange,false);
    this._NgFor_4_6.ngDoCheck(this,this._anchor_4,throwOnChange);
    this._vc_4.detectChangesInNestedViews(throwOnChange);
  }
  destroyInternal():void {
    this._vc_4.destroyNestedViews();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  createEmbeddedViewInternal(nodeIndex:number):import2.AppView<any> {
    if ((nodeIndex == 4)) { return new View_DayViewComponent11(this.viewUtils,this,4,this._anchor_4,this._vc_4); }
    return (null as any);
  }
}
class View_DayViewComponent11 extends import2.AppView<any> {
  _el_0:any;
  _text_1:any;
  _el_2:any;
  _text_3:any;
  _text_4:any;
  _el_5:any;
  _text_6:any;
  _text_7:any;
  /*private*/ _expr_8:any;
  _pipe_date_0_1:any;
  constructor(viewUtils:import3.ViewUtils,parentView:import2.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import16.ViewContainer) {
    super(View_DayViewComponent11,renderType_DayViewComponent,import6.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import1.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._expr_8 = import1.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import7.ComponentRef<any> {
    this._el_0 = import3.createRenderElement(this.renderer,(null as any),'tr',import3.EMPTY_INLINE_ARRAY,(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'\n                            ',(null as any));
    this._el_2 = import3.createRenderElement(this.renderer,this._el_0,'td',new import3.InlineArray2(2,'class','calendar-hour-column text-center'),(null as any));
    this._text_3 = this.renderer.createText(this._el_2,'',(null as any));
    this._text_4 = this.renderer.createText(this._el_0,'\n                            ',(null as any));
    this._el_5 = import3.createRenderElement(this.renderer,this._el_0,'td',new import3.InlineArray2(2,'class','calendar-cell'),(null as any));
    this._text_6 = this.renderer.createText(this._el_5,'\n                            ',(null as any));
    this._text_7 = this.renderer.createText(this._el_0,'\n                        ',(null as any));
    this._pipe_date_0_1 = import3.pureProxy2((<View_DayViewComponent0>this.parentView.parentView)._pipe_date_0.transform.bind((<View_DayViewComponent0>this.parentView.parentView)._pipe_date_0));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._el_2,
      this._text_3,
      this._text_4,
      this._el_5,
      this._text_6,
      this._text_7
    ]
    ),(null as any));
    return (null as any);
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const valUnwrapper:any = new import1.ValueUnwrapper();
    valUnwrapper.reset();
    const currVal_8:any = import3.inlineInterpolate(1,'\n                                ',valUnwrapper.unwrap(import3.castByValue(this._pipe_date_0_1,(<View_DayViewComponent0>this.parentView.parentView)._pipe_date_0.transform)(this.context.$implicit.time,this.parentView.parentView.context.formatHourColumn)),'\n                            ');
    if ((valUnwrapper.hasWrappedValue || import3.checkBinding(throwOnChange,this._expr_8,currVal_8))) {
      this.renderer.setText(this._text_3,currVal_8);
      this._expr_8 = currVal_8;
    }
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
class View_DayViewComponent12 extends import2.AppView<any> {
  _el_0:any;
  _NgClass_0_3:import27.Wrapper_NgClass;
  _NgStyle_0_4:import28.Wrapper_NgStyle;
  _text_1:any;
  _anchor_2:any;
  /*private*/ _vc_2:import16.ViewContainer;
  _TemplateRef_2_5:any;
  _NgFor_2_6:import29.Wrapper_NgFor;
  _text_3:any;
  _map_9:any;
  _map_10:any;
  constructor(viewUtils:import3.ViewUtils,parentView:import2.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import16.ViewContainer) {
    super(View_DayViewComponent12,renderType_DayViewComponent,import6.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import1.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._map_9 = import3.pureProxy1((p0:any):{[key: string]:any} => {
      return {'calendar-event-wrap': p0};
    });
    this._map_10 = import3.pureProxy1((p0:any):{[key: string]:any} => {
      return {height: p0};
    });
  }
  createInternal(rootSelector:string):import7.ComponentRef<any> {
    this._el_0 = import3.createRenderElement(this.renderer,(null as any),'td',new import3.InlineArray2(2,'class','calendar-cell'),(null as any));
    this._NgClass_0_3 = new import27.Wrapper_NgClass(this.parentView.parentView.injectorGet(import30.IterableDiffers,this.parentView.parentIndex),this.parentView.parentView.injectorGet(import31.KeyValueDiffers,this.parentView.parentIndex),new import23.ElementRef(this._el_0),this.renderer);
    this._NgStyle_0_4 = new import28.Wrapper_NgStyle(this.parentView.parentView.injectorGet(import31.KeyValueDiffers,this.parentView.parentIndex),new import23.ElementRef(this._el_0),this.renderer);
    this._text_1 = this.renderer.createText(this._el_0,'\n                                    ',(null as any));
    this._anchor_2 = this.renderer.createTemplateAnchor(this._el_0,(null as any));
    this._vc_2 = new import16.ViewContainer(2,0,this,this._anchor_2);
    this._TemplateRef_2_5 = new import24.TemplateRef_(this,2,this._anchor_2);
    this._NgFor_2_6 = new import29.Wrapper_NgFor(this._vc_2.vcRef,this._TemplateRef_2_5,this.parentView.parentView.injectorGet(import30.IterableDiffers,this.parentView.parentIndex),this.parentView.ref);
    this._text_3 = this.renderer.createText(this._el_0,'\n                                ',(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._anchor_2,
      this._text_3
    ]
    ),(null as any));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import24.TemplateRef) && (2 === requestNodeIndex))) { return this._TemplateRef_2_5; }
    if (((token === import32.NgFor) && (2 === requestNodeIndex))) { return this._NgFor_2_6.context; }
    if (((token === import33.NgClass) && ((0 <= requestNodeIndex) && (requestNodeIndex <= 3)))) { return this._NgClass_0_3.context; }
    if (((token === import34.NgStyle) && ((0 <= requestNodeIndex) && (requestNodeIndex <= 3)))) { return this._NgStyle_0_4.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_0_0_0:any = 'calendar-cell';
    this._NgClass_0_3.check_klass(currVal_0_0_0,throwOnChange,false);
    const currVal_0_0_1:any = this._map_9((this.parentView.context.views[1].allDayEvents.length > 0));
    this._NgClass_0_3.check_ngClass(currVal_0_0_1,throwOnChange,false);
    this._NgClass_0_3.ngDoCheck(this,this._el_0,throwOnChange);
    const currVal_0_1_0:any = this._map_10(((25 * this.parentView.context.views[1].allDayEvents.length) + 'px'));
    this._NgStyle_0_4.check_ngStyle(currVal_0_1_0,throwOnChange,false);
    this._NgStyle_0_4.ngDoCheck(this,this._el_0,throwOnChange);
    const currVal_2_0_0:any = this.parentView.context.views[1].allDayEvents;
    this._NgFor_2_6.check_ngForOf(currVal_2_0_0,throwOnChange,false);
    this._NgFor_2_6.ngDoCheck(this,this._anchor_2,throwOnChange);
    this._vc_2.detectChangesInNestedViews(throwOnChange);
  }
  destroyInternal():void {
    this._vc_2.destroyNestedViews();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  createEmbeddedViewInternal(nodeIndex:number):import2.AppView<any> {
    if ((nodeIndex == 2)) { return new View_DayViewComponent13(this.viewUtils,this,2,this._anchor_2,this._vc_2); }
    return (null as any);
  }
}
class View_DayViewComponent13 extends import2.AppView<any> {
  _el_0:any;
  _NgStyle_0_3:import28.Wrapper_NgStyle;
  _text_1:any;
  _anchor_2:any;
  /*private*/ _vc_2:import16.ViewContainer;
  _TemplateRef_2_5:any;
  _NgTemplateOutlet_2_6:import35.Wrapper_NgTemplateOutlet;
  _text_3:any;
  _map_8:any;
  _map_9:any;
  constructor(viewUtils:import3.ViewUtils,parentView:import2.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import16.ViewContainer) {
    super(View_DayViewComponent13,renderType_DayViewComponent,import6.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import1.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._map_8 = import3.pureProxy3((p0:any,p1:any,p2:any):{[key: string]:any} => {
      return {
        top: p0,
        width: p1,
        height: p2
      }
      ;
    });
    this._map_9 = import3.pureProxy1((p0:any):{[key: string]:any} => {
      return {displayEvent: p0};
    });
  }
  createInternal(rootSelector:string):import7.ComponentRef<any> {
    this._el_0 = import3.createRenderElement(this.renderer,(null as any),'div',new import3.InlineArray4(4,'class','calendar-event','tappable',''),(null as any));
    this._NgStyle_0_3 = new import28.Wrapper_NgStyle(this.parentView.parentView.parentView.injectorGet(import31.KeyValueDiffers,this.parentView.parentView.parentIndex),new import23.ElementRef(this._el_0),this.renderer);
    this._text_1 = this.renderer.createText(this._el_0,'\n                                        ',(null as any));
    this._anchor_2 = this.renderer.createTemplateAnchor(this._el_0,(null as any));
    this._vc_2 = new import16.ViewContainer(2,0,this,this._anchor_2);
    this._TemplateRef_2_5 = new import24.TemplateRef_(this,2,this._anchor_2);
    this._NgTemplateOutlet_2_6 = new import35.Wrapper_NgTemplateOutlet(this._vc_2.vcRef);
    this._text_3 = this.renderer.createText(this._el_0,'\n                                    ',(null as any));
    var disposable_0:Function = import3.subscribeToRenderElement(this,this._el_0,new import3.InlineArray2(2,'click',(null as any)),this.eventHandler(this.handleEvent_0));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._anchor_2,
      this._text_3
    ]
    ),[disposable_0]);
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import24.TemplateRef) && (2 === requestNodeIndex))) { return this._TemplateRef_2_5; }
    if (((token === import36.NgTemplateOutlet) && (2 === requestNodeIndex))) { return this._NgTemplateOutlet_2_6.context; }
    if (((token === import34.NgStyle) && ((0 <= requestNodeIndex) && (requestNodeIndex <= 3)))) { return this._NgStyle_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_0_0_0:any = this._map_8(((25 * this.context.index) + 'px'),'100%','25px');
    this._NgStyle_0_3.check_ngStyle(currVal_0_0_0,throwOnChange,false);
    this._NgStyle_0_3.ngDoCheck(this,this._el_0,throwOnChange);
    const currVal_2_0_0:any = this._map_9(this.context.$implicit);
    this._NgTemplateOutlet_2_6.check_ngOutletContext(currVal_2_0_0,throwOnChange,false);
    const currVal_2_0_1:any = this.parentView.parentView.context.dayviewAllDayEventTemplate;
    this._NgTemplateOutlet_2_6.check_ngTemplateOutlet(currVal_2_0_1,throwOnChange,false);
    this._NgTemplateOutlet_2_6.ngDoCheck(this,this._anchor_2,throwOnChange);
    this._vc_2.detectChangesInNestedViews(throwOnChange);
  }
  destroyInternal():void {
    this._vc_2.destroyNestedViews();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  createEmbeddedViewInternal(nodeIndex:number):import2.AppView<any> {
    if ((nodeIndex == 2)) { return new View_DayViewComponent14(this.viewUtils,this,2,this._anchor_2,this._vc_2); }
    return (null as any);
  }
  handleEvent_0(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.parentView.parentView.context.eventSelected(this.context.$implicit.event)) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
}
class View_DayViewComponent14 extends import2.AppView<any> {
  _text_0:any;
  constructor(viewUtils:import3.ViewUtils,parentView:import2.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import16.ViewContainer) {
    super(View_DayViewComponent14,renderType_DayViewComponent,import6.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import1.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
  }
  createInternal(rootSelector:string):import7.ComponentRef<any> {
    this._text_0 = this.renderer.createText((null as any),'\n                                        ',(null as any));
    this.init(this._text_0,((<any>this.renderer).directRenderer? (null as any): [this._text_0]),(null as any));
    return (null as any);
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._text_0,ctx);
  }
}
class View_DayViewComponent15 extends import2.AppView<any> {
  _el_0:any;
  _text_1:any;
  constructor(viewUtils:import3.ViewUtils,parentView:import2.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import16.ViewContainer) {
    super(View_DayViewComponent15,renderType_DayViewComponent,import6.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import1.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
  }
  createInternal(rootSelector:string):import7.ComponentRef<any> {
    this._el_0 = import3.createRenderElement(this.renderer,(null as any),'td',new import3.InlineArray2(2,'class','calendar-cell'),(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'\n                                ',(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1
    ]
    ),(null as any));
    return (null as any);
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
class View_DayViewComponent16 extends import2.AppView<any> {
  _el_0:any;
  _text_1:any;
  _el_2:any;
  _text_3:any;
  _anchor_4:any;
  /*private*/ _vc_4:import16.ViewContainer;
  _TemplateRef_4_5:any;
  _NgFor_4_6:import29.Wrapper_NgFor;
  _text_5:any;
  _text_6:any;
  constructor(viewUtils:import3.ViewUtils,parentView:import2.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import16.ViewContainer) {
    super(View_DayViewComponent16,renderType_DayViewComponent,import6.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import1.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
  }
  createInternal(rootSelector:string):import7.ComponentRef<any> {
    this._el_0 = import3.createRenderElement(this.renderer,(null as any),'table',new import3.InlineArray2(2,'class','table table-bordered table-fixed dayview-normal-event-table'),(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'\n                        ',(null as any));
    this._el_2 = import3.createRenderElement(this.renderer,this._el_0,'tbody',import3.EMPTY_INLINE_ARRAY,(null as any));
    this._text_3 = this.renderer.createText(this._el_2,'\n                        ',(null as any));
    this._anchor_4 = this.renderer.createTemplateAnchor(this._el_2,(null as any));
    this._vc_4 = new import16.ViewContainer(4,2,this,this._anchor_4);
    this._TemplateRef_4_5 = new import24.TemplateRef_(this,4,this._anchor_4);
    this._NgFor_4_6 = new import29.Wrapper_NgFor(this._vc_4.vcRef,this._TemplateRef_4_5,this.parentView.parentView.injectorGet(import30.IterableDiffers,this.parentView.parentIndex),this.parentView.ref);
    this._text_5 = this.renderer.createText(this._el_2,'\n                        ',(null as any));
    this._text_6 = this.renderer.createText(this._el_0,'\n                    ',(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._el_2,
      this._text_3,
      this._anchor_4,
      this._text_5,
      this._text_6
    ]
    ),(null as any));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import24.TemplateRef) && (4 === requestNodeIndex))) { return this._TemplateRef_4_5; }
    if (((token === import32.NgFor) && (4 === requestNodeIndex))) { return this._NgFor_4_6.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_4_0_0:any = this.parentView.context.views[1].rows;
    this._NgFor_4_6.check_ngForOf(currVal_4_0_0,throwOnChange,false);
    this._NgFor_4_6.ngDoCheck(this,this._anchor_4,throwOnChange);
    this._vc_4.detectChangesInNestedViews(throwOnChange);
  }
  destroyInternal():void {
    this._vc_4.destroyNestedViews();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  createEmbeddedViewInternal(nodeIndex:number):import2.AppView<any> {
    if ((nodeIndex == 4)) { return new View_DayViewComponent17(this.viewUtils,this,4,this._anchor_4,this._vc_4); }
    return (null as any);
  }
}
class View_DayViewComponent17 extends import2.AppView<any> {
  _el_0:any;
  _text_1:any;
  _el_2:any;
  _text_3:any;
  _text_4:any;
  _el_5:any;
  _text_6:any;
  _anchor_7:any;
  /*private*/ _vc_7:import16.ViewContainer;
  _TemplateRef_7_5:any;
  _NgIf_7_6:import17.Wrapper_NgIf;
  _text_8:any;
  _text_9:any;
  /*private*/ _expr_13:any;
  _pipe_date_0_2:any;
  constructor(viewUtils:import3.ViewUtils,parentView:import2.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import16.ViewContainer) {
    super(View_DayViewComponent17,renderType_DayViewComponent,import6.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import1.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._expr_13 = import1.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import7.ComponentRef<any> {
    this._el_0 = import3.createRenderElement(this.renderer,(null as any),'tr',import3.EMPTY_INLINE_ARRAY,(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'\n                            ',(null as any));
    this._el_2 = import3.createRenderElement(this.renderer,this._el_0,'td',new import3.InlineArray2(2,'class','calendar-hour-column text-center'),(null as any));
    this._text_3 = this.renderer.createText(this._el_2,'',(null as any));
    this._text_4 = this.renderer.createText(this._el_0,'\n                            ',(null as any));
    this._el_5 = import3.createRenderElement(this.renderer,this._el_0,'td',new import3.InlineArray4(4,'class','calendar-cell','tappable',''),(null as any));
    this._text_6 = this.renderer.createText(this._el_5,'\n                                ',(null as any));
    this._anchor_7 = this.renderer.createTemplateAnchor(this._el_5,(null as any));
    this._vc_7 = new import16.ViewContainer(7,5,this,this._anchor_7);
    this._TemplateRef_7_5 = new import24.TemplateRef_(this,7,this._anchor_7);
    this._NgIf_7_6 = new import17.Wrapper_NgIf(this._vc_7.vcRef,this._TemplateRef_7_5);
    this._text_8 = this.renderer.createText(this._el_5,'\n                            ',(null as any));
    this._text_9 = this.renderer.createText(this._el_0,'\n                        ',(null as any));
    this._pipe_date_0_2 = import3.pureProxy2((<View_DayViewComponent0>this.parentView.parentView)._pipe_date_0.transform.bind((<View_DayViewComponent0>this.parentView.parentView)._pipe_date_0));
    var disposable_0:Function = import3.subscribeToRenderElement(this,this._el_5,new import3.InlineArray2(2,'click',(null as any)),this.eventHandler(this.handleEvent_5));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._el_2,
      this._text_3,
      this._text_4,
      this._el_5,
      this._text_6,
      this._anchor_7,
      this._text_8,
      this._text_9
    ]
    ),[disposable_0]);
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import24.TemplateRef) && (7 === requestNodeIndex))) { return this._TemplateRef_7_5; }
    if (((token === import26.NgIf) && (7 === requestNodeIndex))) { return this._NgIf_7_6.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const valUnwrapper:any = new import1.ValueUnwrapper();
    const currVal_7_0_0:any = this.context.$implicit.events;
    this._NgIf_7_6.check_ngIf(currVal_7_0_0,throwOnChange,false);
    this._NgIf_7_6.ngDoCheck(this,this._anchor_7,throwOnChange);
    this._vc_7.detectChangesInNestedViews(throwOnChange);
    valUnwrapper.reset();
    const currVal_13:any = import3.inlineInterpolate(1,'\n                                ',valUnwrapper.unwrap(import3.castByValue(this._pipe_date_0_2,(<View_DayViewComponent0>this.parentView.parentView)._pipe_date_0.transform)(this.context.$implicit.time,this.parentView.parentView.context.formatHourColumn)),'\n                            ');
    if ((valUnwrapper.hasWrappedValue || import3.checkBinding(throwOnChange,this._expr_13,currVal_13))) {
      this.renderer.setText(this._text_3,currVal_13);
      this._expr_13 = currVal_13;
    }
  }
  destroyInternal():void {
    this._vc_7.destroyNestedViews();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  createEmbeddedViewInternal(nodeIndex:number):import2.AppView<any> {
    if ((nodeIndex == 7)) { return new View_DayViewComponent18(this.viewUtils,this,7,this._anchor_7,this._vc_7); }
    return (null as any);
  }
  handleEvent_5(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.parentView.parentView.context.select(this.context.$implicit.time,this.context.$implicit.events)) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
}
class View_DayViewComponent18 extends import2.AppView<any> {
  _el_0:any;
  _NgClass_0_3:import27.Wrapper_NgClass;
  _text_1:any;
  _anchor_2:any;
  /*private*/ _vc_2:import16.ViewContainer;
  _TemplateRef_2_5:any;
  _NgFor_2_6:import29.Wrapper_NgFor;
  _text_3:any;
  _map_8:any;
  constructor(viewUtils:import3.ViewUtils,parentView:import2.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import16.ViewContainer) {
    super(View_DayViewComponent18,renderType_DayViewComponent,import6.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import1.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._map_8 = import3.pureProxy1((p0:any):{[key: string]:any} => {
      return {'calendar-event-wrap': p0};
    });
  }
  createInternal(rootSelector:string):import7.ComponentRef<any> {
    this._el_0 = import3.createRenderElement(this.renderer,(null as any),'div',import3.EMPTY_INLINE_ARRAY,(null as any));
    this._NgClass_0_3 = new import27.Wrapper_NgClass(this.parentView.parentView.parentView.parentView.injectorGet(import30.IterableDiffers,this.parentView.parentView.parentView.parentIndex),this.parentView.parentView.parentView.parentView.injectorGet(import31.KeyValueDiffers,this.parentView.parentView.parentView.parentIndex),new import23.ElementRef(this._el_0),this.renderer);
    this._text_1 = this.renderer.createText(this._el_0,'\n                                    ',(null as any));
    this._anchor_2 = this.renderer.createTemplateAnchor(this._el_0,(null as any));
    this._vc_2 = new import16.ViewContainer(2,0,this,this._anchor_2);
    this._TemplateRef_2_5 = new import24.TemplateRef_(this,2,this._anchor_2);
    this._NgFor_2_6 = new import29.Wrapper_NgFor(this._vc_2.vcRef,this._TemplateRef_2_5,this.parentView.parentView.parentView.parentView.injectorGet(import30.IterableDiffers,this.parentView.parentView.parentView.parentIndex),this.parentView.parentView.parentView.ref);
    this._text_3 = this.renderer.createText(this._el_0,'\n                                ',(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._anchor_2,
      this._text_3
    ]
    ),(null as any));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import24.TemplateRef) && (2 === requestNodeIndex))) { return this._TemplateRef_2_5; }
    if (((token === import32.NgFor) && (2 === requestNodeIndex))) { return this._NgFor_2_6.context; }
    if (((token === import33.NgClass) && ((0 <= requestNodeIndex) && (requestNodeIndex <= 3)))) { return this._NgClass_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_0_0_0:any = this._map_8(this.parentView.context.$implicit.events);
    this._NgClass_0_3.check_ngClass(currVal_0_0_0,throwOnChange,false);
    this._NgClass_0_3.ngDoCheck(this,this._el_0,throwOnChange);
    const currVal_2_0_0:any = this.parentView.context.$implicit.events;
    this._NgFor_2_6.check_ngForOf(currVal_2_0_0,throwOnChange,false);
    this._NgFor_2_6.ngDoCheck(this,this._anchor_2,throwOnChange);
    this._vc_2.detectChangesInNestedViews(throwOnChange);
  }
  destroyInternal():void {
    this._vc_2.destroyNestedViews();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  createEmbeddedViewInternal(nodeIndex:number):import2.AppView<any> {
    if ((nodeIndex == 2)) { return new View_DayViewComponent19(this.viewUtils,this,2,this._anchor_2,this._vc_2); }
    return (null as any);
  }
}
class View_DayViewComponent19 extends import2.AppView<any> {
  _el_0:any;
  _NgStyle_0_3:import28.Wrapper_NgStyle;
  _text_1:any;
  _anchor_2:any;
  /*private*/ _vc_2:import16.ViewContainer;
  _TemplateRef_2_5:any;
  _NgTemplateOutlet_2_6:import35.Wrapper_NgTemplateOutlet;
  _text_3:any;
  _map_8:any;
  _map_9:any;
  constructor(viewUtils:import3.ViewUtils,parentView:import2.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import16.ViewContainer) {
    super(View_DayViewComponent19,renderType_DayViewComponent,import6.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import1.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._map_8 = import3.pureProxy4((p0:any,p1:any,p2:any,p3:any):{[key: string]:any} => {
      return {
        top: p0,
        left: p1,
        width: p2,
        height: p3
      }
      ;
    });
    this._map_9 = import3.pureProxy1((p0:any):{[key: string]:any} => {
      return {displayEvent: p0};
    });
  }
  createInternal(rootSelector:string):import7.ComponentRef<any> {
    this._el_0 = import3.createRenderElement(this.renderer,(null as any),'div',new import3.InlineArray4(4,'class','calendar-event','tappable',''),(null as any));
    this._NgStyle_0_3 = new import28.Wrapper_NgStyle(this.parentView.parentView.parentView.parentView.parentView.injectorGet(import31.KeyValueDiffers,this.parentView.parentView.parentView.parentView.parentIndex),new import23.ElementRef(this._el_0),this.renderer);
    this._text_1 = this.renderer.createText(this._el_0,'\n                                         ',(null as any));
    this._anchor_2 = this.renderer.createTemplateAnchor(this._el_0,(null as any));
    this._vc_2 = new import16.ViewContainer(2,0,this,this._anchor_2);
    this._TemplateRef_2_5 = new import24.TemplateRef_(this,2,this._anchor_2);
    this._NgTemplateOutlet_2_6 = new import35.Wrapper_NgTemplateOutlet(this._vc_2.vcRef);
    this._text_3 = this.renderer.createText(this._el_0,'\n                                    ',(null as any));
    var disposable_0:Function = import3.subscribeToRenderElement(this,this._el_0,new import3.InlineArray2(2,'click',(null as any)),this.eventHandler(this.handleEvent_0));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._anchor_2,
      this._text_3
    ]
    ),[disposable_0]);
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import24.TemplateRef) && (2 === requestNodeIndex))) { return this._TemplateRef_2_5; }
    if (((token === import36.NgTemplateOutlet) && (2 === requestNodeIndex))) { return this._NgTemplateOutlet_2_6.context; }
    if (((token === import34.NgStyle) && ((0 <= requestNodeIndex) && (requestNodeIndex <= 3)))) { return this._NgStyle_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_0_0_0:any = this._map_8((((37 * this.context.$implicit.startOffset) / this.parentView.parentView.parentView.parentView.context.hourParts) + 'px'),(((100 / this.context.$implicit.overlapNumber) * this.context.$implicit.position) + '%'),((100 / this.context.$implicit.overlapNumber) + '%'),((37 * ((this.context.$implicit.endIndex - this.context.$implicit.startIndex) - ((this.context.$implicit.endOffset + this.context.$implicit.startOffset) / this.parentView.parentView.parentView.parentView.context.hourParts))) + 'px'));
    this._NgStyle_0_3.check_ngStyle(currVal_0_0_0,throwOnChange,false);
    this._NgStyle_0_3.ngDoCheck(this,this._el_0,throwOnChange);
    const currVal_2_0_0:any = this._map_9(this.context.$implicit);
    this._NgTemplateOutlet_2_6.check_ngOutletContext(currVal_2_0_0,throwOnChange,false);
    const currVal_2_0_1:any = this.parentView.parentView.parentView.parentView.context.dayviewNormalEventTemplate;
    this._NgTemplateOutlet_2_6.check_ngTemplateOutlet(currVal_2_0_1,throwOnChange,false);
    this._NgTemplateOutlet_2_6.ngDoCheck(this,this._anchor_2,throwOnChange);
    this._vc_2.detectChangesInNestedViews(throwOnChange);
  }
  destroyInternal():void {
    this._vc_2.destroyNestedViews();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  createEmbeddedViewInternal(nodeIndex:number):import2.AppView<any> {
    if ((nodeIndex == 2)) { return new View_DayViewComponent20(this.viewUtils,this,2,this._anchor_2,this._vc_2); }
    return (null as any);
  }
  handleEvent_0(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.parentView.parentView.parentView.parentView.context.eventSelected(this.context.$implicit.event)) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
}
class View_DayViewComponent20 extends import2.AppView<any> {
  _text_0:any;
  constructor(viewUtils:import3.ViewUtils,parentView:import2.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import16.ViewContainer) {
    super(View_DayViewComponent20,renderType_DayViewComponent,import6.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import1.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
  }
  createInternal(rootSelector:string):import7.ComponentRef<any> {
    this._text_0 = this.renderer.createText((null as any),'\n                                         ',(null as any));
    this.init(this._text_0,((<any>this.renderer).directRenderer? (null as any): [this._text_0]),(null as any));
    return (null as any);
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._text_0,ctx);
  }
}
class View_DayViewComponent21 extends import2.AppView<any> {
  _el_0:any;
  _text_1:any;
  _el_2:any;
  _text_3:any;
  _anchor_4:any;
  /*private*/ _vc_4:import16.ViewContainer;
  _TemplateRef_4_5:any;
  _NgFor_4_6:import29.Wrapper_NgFor;
  _text_5:any;
  _text_6:any;
  constructor(viewUtils:import3.ViewUtils,parentView:import2.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import16.ViewContainer) {
    super(View_DayViewComponent21,renderType_DayViewComponent,import6.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import1.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
  }
  createInternal(rootSelector:string):import7.ComponentRef<any> {
    this._el_0 = import3.createRenderElement(this.renderer,(null as any),'table',new import3.InlineArray2(2,'class','table table-bordered table-fixed dayview-normal-event-table'),(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'\n                        ',(null as any));
    this._el_2 = import3.createRenderElement(this.renderer,this._el_0,'tbody',import3.EMPTY_INLINE_ARRAY,(null as any));
    this._text_3 = this.renderer.createText(this._el_2,'\n                        ',(null as any));
    this._anchor_4 = this.renderer.createTemplateAnchor(this._el_2,(null as any));
    this._vc_4 = new import16.ViewContainer(4,2,this,this._anchor_4);
    this._TemplateRef_4_5 = new import24.TemplateRef_(this,4,this._anchor_4);
    this._NgFor_4_6 = new import29.Wrapper_NgFor(this._vc_4.vcRef,this._TemplateRef_4_5,this.parentView.parentView.injectorGet(import30.IterableDiffers,this.parentView.parentIndex),this.parentView.ref);
    this._text_5 = this.renderer.createText(this._el_2,'\n                        ',(null as any));
    this._text_6 = this.renderer.createText(this._el_0,'\n                    ',(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._el_2,
      this._text_3,
      this._anchor_4,
      this._text_5,
      this._text_6
    ]
    ),(null as any));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import24.TemplateRef) && (4 === requestNodeIndex))) { return this._TemplateRef_4_5; }
    if (((token === import32.NgFor) && (4 === requestNodeIndex))) { return this._NgFor_4_6.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_4_0_0:any = this.parentView.context.views[1].rows;
    this._NgFor_4_6.check_ngForOf(currVal_4_0_0,throwOnChange,false);
    this._NgFor_4_6.ngDoCheck(this,this._anchor_4,throwOnChange);
    this._vc_4.detectChangesInNestedViews(throwOnChange);
  }
  destroyInternal():void {
    this._vc_4.destroyNestedViews();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  createEmbeddedViewInternal(nodeIndex:number):import2.AppView<any> {
    if ((nodeIndex == 4)) { return new View_DayViewComponent22(this.viewUtils,this,4,this._anchor_4,this._vc_4); }
    return (null as any);
  }
}
class View_DayViewComponent22 extends import2.AppView<any> {
  _el_0:any;
  _text_1:any;
  _el_2:any;
  _text_3:any;
  _text_4:any;
  _el_5:any;
  _text_6:any;
  _text_7:any;
  /*private*/ _expr_8:any;
  _pipe_date_0_3:any;
  constructor(viewUtils:import3.ViewUtils,parentView:import2.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import16.ViewContainer) {
    super(View_DayViewComponent22,renderType_DayViewComponent,import6.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import1.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._expr_8 = import1.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import7.ComponentRef<any> {
    this._el_0 = import3.createRenderElement(this.renderer,(null as any),'tr',import3.EMPTY_INLINE_ARRAY,(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'\n                            ',(null as any));
    this._el_2 = import3.createRenderElement(this.renderer,this._el_0,'td',new import3.InlineArray2(2,'class','calendar-hour-column text-center'),(null as any));
    this._text_3 = this.renderer.createText(this._el_2,'',(null as any));
    this._text_4 = this.renderer.createText(this._el_0,'\n                            ',(null as any));
    this._el_5 = import3.createRenderElement(this.renderer,this._el_0,'td',new import3.InlineArray2(2,'class','calendar-cell'),(null as any));
    this._text_6 = this.renderer.createText(this._el_5,'\n                            ',(null as any));
    this._text_7 = this.renderer.createText(this._el_0,'\n                        ',(null as any));
    this._pipe_date_0_3 = import3.pureProxy2((<View_DayViewComponent0>this.parentView.parentView)._pipe_date_0.transform.bind((<View_DayViewComponent0>this.parentView.parentView)._pipe_date_0));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._el_2,
      this._text_3,
      this._text_4,
      this._el_5,
      this._text_6,
      this._text_7
    ]
    ),(null as any));
    return (null as any);
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const valUnwrapper:any = new import1.ValueUnwrapper();
    valUnwrapper.reset();
    const currVal_8:any = import3.inlineInterpolate(1,'\n                                ',valUnwrapper.unwrap(import3.castByValue(this._pipe_date_0_3,(<View_DayViewComponent0>this.parentView.parentView)._pipe_date_0.transform)(this.context.$implicit.time,this.parentView.parentView.context.formatHourColumn)),'\n                            ');
    if ((valUnwrapper.hasWrappedValue || import3.checkBinding(throwOnChange,this._expr_8,currVal_8))) {
      this.renderer.setText(this._text_3,currVal_8);
      this._expr_8 = currVal_8;
    }
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
class View_DayViewComponent23 extends import2.AppView<any> {
  _el_0:any;
  _NgClass_0_3:import27.Wrapper_NgClass;
  _NgStyle_0_4:import28.Wrapper_NgStyle;
  _text_1:any;
  _anchor_2:any;
  /*private*/ _vc_2:import16.ViewContainer;
  _TemplateRef_2_5:any;
  _NgFor_2_6:import29.Wrapper_NgFor;
  _text_3:any;
  _map_9:any;
  _map_10:any;
  constructor(viewUtils:import3.ViewUtils,parentView:import2.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import16.ViewContainer) {
    super(View_DayViewComponent23,renderType_DayViewComponent,import6.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import1.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._map_9 = import3.pureProxy1((p0:any):{[key: string]:any} => {
      return {'calendar-event-wrap': p0};
    });
    this._map_10 = import3.pureProxy1((p0:any):{[key: string]:any} => {
      return {height: p0};
    });
  }
  createInternal(rootSelector:string):import7.ComponentRef<any> {
    this._el_0 = import3.createRenderElement(this.renderer,(null as any),'td',new import3.InlineArray2(2,'class','calendar-cell'),(null as any));
    this._NgClass_0_3 = new import27.Wrapper_NgClass(this.parentView.parentView.injectorGet(import30.IterableDiffers,this.parentView.parentIndex),this.parentView.parentView.injectorGet(import31.KeyValueDiffers,this.parentView.parentIndex),new import23.ElementRef(this._el_0),this.renderer);
    this._NgStyle_0_4 = new import28.Wrapper_NgStyle(this.parentView.parentView.injectorGet(import31.KeyValueDiffers,this.parentView.parentIndex),new import23.ElementRef(this._el_0),this.renderer);
    this._text_1 = this.renderer.createText(this._el_0,'\n                                    ',(null as any));
    this._anchor_2 = this.renderer.createTemplateAnchor(this._el_0,(null as any));
    this._vc_2 = new import16.ViewContainer(2,0,this,this._anchor_2);
    this._TemplateRef_2_5 = new import24.TemplateRef_(this,2,this._anchor_2);
    this._NgFor_2_6 = new import29.Wrapper_NgFor(this._vc_2.vcRef,this._TemplateRef_2_5,this.parentView.parentView.injectorGet(import30.IterableDiffers,this.parentView.parentIndex),this.parentView.ref);
    this._text_3 = this.renderer.createText(this._el_0,'\n                                ',(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._anchor_2,
      this._text_3
    ]
    ),(null as any));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import24.TemplateRef) && (2 === requestNodeIndex))) { return this._TemplateRef_2_5; }
    if (((token === import32.NgFor) && (2 === requestNodeIndex))) { return this._NgFor_2_6.context; }
    if (((token === import33.NgClass) && ((0 <= requestNodeIndex) && (requestNodeIndex <= 3)))) { return this._NgClass_0_3.context; }
    if (((token === import34.NgStyle) && ((0 <= requestNodeIndex) && (requestNodeIndex <= 3)))) { return this._NgStyle_0_4.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_0_0_0:any = 'calendar-cell';
    this._NgClass_0_3.check_klass(currVal_0_0_0,throwOnChange,false);
    const currVal_0_0_1:any = this._map_9((this.parentView.context.views[2].allDayEvents.length > 0));
    this._NgClass_0_3.check_ngClass(currVal_0_0_1,throwOnChange,false);
    this._NgClass_0_3.ngDoCheck(this,this._el_0,throwOnChange);
    const currVal_0_1_0:any = this._map_10(((25 * this.parentView.context.views[2].allDayEvents.length) + 'px'));
    this._NgStyle_0_4.check_ngStyle(currVal_0_1_0,throwOnChange,false);
    this._NgStyle_0_4.ngDoCheck(this,this._el_0,throwOnChange);
    const currVal_2_0_0:any = this.parentView.context.views[2].allDayEvents;
    this._NgFor_2_6.check_ngForOf(currVal_2_0_0,throwOnChange,false);
    this._NgFor_2_6.ngDoCheck(this,this._anchor_2,throwOnChange);
    this._vc_2.detectChangesInNestedViews(throwOnChange);
  }
  destroyInternal():void {
    this._vc_2.destroyNestedViews();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  createEmbeddedViewInternal(nodeIndex:number):import2.AppView<any> {
    if ((nodeIndex == 2)) { return new View_DayViewComponent24(this.viewUtils,this,2,this._anchor_2,this._vc_2); }
    return (null as any);
  }
}
class View_DayViewComponent24 extends import2.AppView<any> {
  _el_0:any;
  _NgStyle_0_3:import28.Wrapper_NgStyle;
  _text_1:any;
  _anchor_2:any;
  /*private*/ _vc_2:import16.ViewContainer;
  _TemplateRef_2_5:any;
  _NgTemplateOutlet_2_6:import35.Wrapper_NgTemplateOutlet;
  _text_3:any;
  _map_8:any;
  _map_9:any;
  constructor(viewUtils:import3.ViewUtils,parentView:import2.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import16.ViewContainer) {
    super(View_DayViewComponent24,renderType_DayViewComponent,import6.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import1.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._map_8 = import3.pureProxy3((p0:any,p1:any,p2:any):{[key: string]:any} => {
      return {
        top: p0,
        width: p1,
        height: p2
      }
      ;
    });
    this._map_9 = import3.pureProxy1((p0:any):{[key: string]:any} => {
      return {displayEvent: p0};
    });
  }
  createInternal(rootSelector:string):import7.ComponentRef<any> {
    this._el_0 = import3.createRenderElement(this.renderer,(null as any),'div',new import3.InlineArray4(4,'class','calendar-event','tappable',''),(null as any));
    this._NgStyle_0_3 = new import28.Wrapper_NgStyle(this.parentView.parentView.parentView.injectorGet(import31.KeyValueDiffers,this.parentView.parentView.parentIndex),new import23.ElementRef(this._el_0),this.renderer);
    this._text_1 = this.renderer.createText(this._el_0,'\n                                        ',(null as any));
    this._anchor_2 = this.renderer.createTemplateAnchor(this._el_0,(null as any));
    this._vc_2 = new import16.ViewContainer(2,0,this,this._anchor_2);
    this._TemplateRef_2_5 = new import24.TemplateRef_(this,2,this._anchor_2);
    this._NgTemplateOutlet_2_6 = new import35.Wrapper_NgTemplateOutlet(this._vc_2.vcRef);
    this._text_3 = this.renderer.createText(this._el_0,'\n                                    ',(null as any));
    var disposable_0:Function = import3.subscribeToRenderElement(this,this._el_0,new import3.InlineArray2(2,'click',(null as any)),this.eventHandler(this.handleEvent_0));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._anchor_2,
      this._text_3
    ]
    ),[disposable_0]);
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import24.TemplateRef) && (2 === requestNodeIndex))) { return this._TemplateRef_2_5; }
    if (((token === import36.NgTemplateOutlet) && (2 === requestNodeIndex))) { return this._NgTemplateOutlet_2_6.context; }
    if (((token === import34.NgStyle) && ((0 <= requestNodeIndex) && (requestNodeIndex <= 3)))) { return this._NgStyle_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_0_0_0:any = this._map_8(((25 * this.context.index) + 'px'),'100%','25px');
    this._NgStyle_0_3.check_ngStyle(currVal_0_0_0,throwOnChange,false);
    this._NgStyle_0_3.ngDoCheck(this,this._el_0,throwOnChange);
    const currVal_2_0_0:any = this._map_9(this.context.$implicit);
    this._NgTemplateOutlet_2_6.check_ngOutletContext(currVal_2_0_0,throwOnChange,false);
    const currVal_2_0_1:any = this.parentView.parentView.context.dayviewAllDayEventTemplate;
    this._NgTemplateOutlet_2_6.check_ngTemplateOutlet(currVal_2_0_1,throwOnChange,false);
    this._NgTemplateOutlet_2_6.ngDoCheck(this,this._anchor_2,throwOnChange);
    this._vc_2.detectChangesInNestedViews(throwOnChange);
  }
  destroyInternal():void {
    this._vc_2.destroyNestedViews();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  createEmbeddedViewInternal(nodeIndex:number):import2.AppView<any> {
    if ((nodeIndex == 2)) { return new View_DayViewComponent25(this.viewUtils,this,2,this._anchor_2,this._vc_2); }
    return (null as any);
  }
  handleEvent_0(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.parentView.parentView.context.eventSelected(this.context.$implicit.event)) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
}
class View_DayViewComponent25 extends import2.AppView<any> {
  _text_0:any;
  constructor(viewUtils:import3.ViewUtils,parentView:import2.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import16.ViewContainer) {
    super(View_DayViewComponent25,renderType_DayViewComponent,import6.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import1.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
  }
  createInternal(rootSelector:string):import7.ComponentRef<any> {
    this._text_0 = this.renderer.createText((null as any),'\n                                        ',(null as any));
    this.init(this._text_0,((<any>this.renderer).directRenderer? (null as any): [this._text_0]),(null as any));
    return (null as any);
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._text_0,ctx);
  }
}
class View_DayViewComponent26 extends import2.AppView<any> {
  _el_0:any;
  _text_1:any;
  constructor(viewUtils:import3.ViewUtils,parentView:import2.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import16.ViewContainer) {
    super(View_DayViewComponent26,renderType_DayViewComponent,import6.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import1.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
  }
  createInternal(rootSelector:string):import7.ComponentRef<any> {
    this._el_0 = import3.createRenderElement(this.renderer,(null as any),'td',new import3.InlineArray2(2,'class','calendar-cell'),(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'\n                                ',(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1
    ]
    ),(null as any));
    return (null as any);
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
class View_DayViewComponent27 extends import2.AppView<any> {
  _el_0:any;
  _text_1:any;
  _el_2:any;
  _text_3:any;
  _anchor_4:any;
  /*private*/ _vc_4:import16.ViewContainer;
  _TemplateRef_4_5:any;
  _NgFor_4_6:import29.Wrapper_NgFor;
  _text_5:any;
  _text_6:any;
  constructor(viewUtils:import3.ViewUtils,parentView:import2.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import16.ViewContainer) {
    super(View_DayViewComponent27,renderType_DayViewComponent,import6.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import1.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
  }
  createInternal(rootSelector:string):import7.ComponentRef<any> {
    this._el_0 = import3.createRenderElement(this.renderer,(null as any),'table',new import3.InlineArray2(2,'class','table table-bordered table-fixed dayview-normal-event-table'),(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'\n                        ',(null as any));
    this._el_2 = import3.createRenderElement(this.renderer,this._el_0,'tbody',import3.EMPTY_INLINE_ARRAY,(null as any));
    this._text_3 = this.renderer.createText(this._el_2,'\n                        ',(null as any));
    this._anchor_4 = this.renderer.createTemplateAnchor(this._el_2,(null as any));
    this._vc_4 = new import16.ViewContainer(4,2,this,this._anchor_4);
    this._TemplateRef_4_5 = new import24.TemplateRef_(this,4,this._anchor_4);
    this._NgFor_4_6 = new import29.Wrapper_NgFor(this._vc_4.vcRef,this._TemplateRef_4_5,this.parentView.parentView.injectorGet(import30.IterableDiffers,this.parentView.parentIndex),this.parentView.ref);
    this._text_5 = this.renderer.createText(this._el_2,'\n                        ',(null as any));
    this._text_6 = this.renderer.createText(this._el_0,'\n                    ',(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._el_2,
      this._text_3,
      this._anchor_4,
      this._text_5,
      this._text_6
    ]
    ),(null as any));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import24.TemplateRef) && (4 === requestNodeIndex))) { return this._TemplateRef_4_5; }
    if (((token === import32.NgFor) && (4 === requestNodeIndex))) { return this._NgFor_4_6.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_4_0_0:any = this.parentView.context.views[2].rows;
    this._NgFor_4_6.check_ngForOf(currVal_4_0_0,throwOnChange,false);
    this._NgFor_4_6.ngDoCheck(this,this._anchor_4,throwOnChange);
    this._vc_4.detectChangesInNestedViews(throwOnChange);
  }
  destroyInternal():void {
    this._vc_4.destroyNestedViews();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  createEmbeddedViewInternal(nodeIndex:number):import2.AppView<any> {
    if ((nodeIndex == 4)) { return new View_DayViewComponent28(this.viewUtils,this,4,this._anchor_4,this._vc_4); }
    return (null as any);
  }
}
class View_DayViewComponent28 extends import2.AppView<any> {
  _el_0:any;
  _text_1:any;
  _el_2:any;
  _text_3:any;
  _text_4:any;
  _el_5:any;
  _text_6:any;
  _anchor_7:any;
  /*private*/ _vc_7:import16.ViewContainer;
  _TemplateRef_7_5:any;
  _NgIf_7_6:import17.Wrapper_NgIf;
  _text_8:any;
  _text_9:any;
  /*private*/ _expr_13:any;
  _pipe_date_0_4:any;
  constructor(viewUtils:import3.ViewUtils,parentView:import2.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import16.ViewContainer) {
    super(View_DayViewComponent28,renderType_DayViewComponent,import6.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import1.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._expr_13 = import1.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import7.ComponentRef<any> {
    this._el_0 = import3.createRenderElement(this.renderer,(null as any),'tr',import3.EMPTY_INLINE_ARRAY,(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'\n                            ',(null as any));
    this._el_2 = import3.createRenderElement(this.renderer,this._el_0,'td',new import3.InlineArray2(2,'class','calendar-hour-column text-center'),(null as any));
    this._text_3 = this.renderer.createText(this._el_2,'',(null as any));
    this._text_4 = this.renderer.createText(this._el_0,'\n                            ',(null as any));
    this._el_5 = import3.createRenderElement(this.renderer,this._el_0,'td',new import3.InlineArray4(4,'class','calendar-cell','tappable',''),(null as any));
    this._text_6 = this.renderer.createText(this._el_5,'\n                                ',(null as any));
    this._anchor_7 = this.renderer.createTemplateAnchor(this._el_5,(null as any));
    this._vc_7 = new import16.ViewContainer(7,5,this,this._anchor_7);
    this._TemplateRef_7_5 = new import24.TemplateRef_(this,7,this._anchor_7);
    this._NgIf_7_6 = new import17.Wrapper_NgIf(this._vc_7.vcRef,this._TemplateRef_7_5);
    this._text_8 = this.renderer.createText(this._el_5,'\n                            ',(null as any));
    this._text_9 = this.renderer.createText(this._el_0,'\n                        ',(null as any));
    this._pipe_date_0_4 = import3.pureProxy2((<View_DayViewComponent0>this.parentView.parentView)._pipe_date_0.transform.bind((<View_DayViewComponent0>this.parentView.parentView)._pipe_date_0));
    var disposable_0:Function = import3.subscribeToRenderElement(this,this._el_5,new import3.InlineArray2(2,'click',(null as any)),this.eventHandler(this.handleEvent_5));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._el_2,
      this._text_3,
      this._text_4,
      this._el_5,
      this._text_6,
      this._anchor_7,
      this._text_8,
      this._text_9
    ]
    ),[disposable_0]);
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import24.TemplateRef) && (7 === requestNodeIndex))) { return this._TemplateRef_7_5; }
    if (((token === import26.NgIf) && (7 === requestNodeIndex))) { return this._NgIf_7_6.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const valUnwrapper:any = new import1.ValueUnwrapper();
    const currVal_7_0_0:any = this.context.$implicit.events;
    this._NgIf_7_6.check_ngIf(currVal_7_0_0,throwOnChange,false);
    this._NgIf_7_6.ngDoCheck(this,this._anchor_7,throwOnChange);
    this._vc_7.detectChangesInNestedViews(throwOnChange);
    valUnwrapper.reset();
    const currVal_13:any = import3.inlineInterpolate(1,'\n                                ',valUnwrapper.unwrap(import3.castByValue(this._pipe_date_0_4,(<View_DayViewComponent0>this.parentView.parentView)._pipe_date_0.transform)(this.context.$implicit.time,this.parentView.parentView.context.formatHourColumn)),'\n                            ');
    if ((valUnwrapper.hasWrappedValue || import3.checkBinding(throwOnChange,this._expr_13,currVal_13))) {
      this.renderer.setText(this._text_3,currVal_13);
      this._expr_13 = currVal_13;
    }
  }
  destroyInternal():void {
    this._vc_7.destroyNestedViews();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  createEmbeddedViewInternal(nodeIndex:number):import2.AppView<any> {
    if ((nodeIndex == 7)) { return new View_DayViewComponent29(this.viewUtils,this,7,this._anchor_7,this._vc_7); }
    return (null as any);
  }
  handleEvent_5(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.parentView.parentView.context.select(this.context.$implicit.time,this.context.$implicit.events)) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
}
class View_DayViewComponent29 extends import2.AppView<any> {
  _el_0:any;
  _NgClass_0_3:import27.Wrapper_NgClass;
  _text_1:any;
  _anchor_2:any;
  /*private*/ _vc_2:import16.ViewContainer;
  _TemplateRef_2_5:any;
  _NgFor_2_6:import29.Wrapper_NgFor;
  _text_3:any;
  _map_8:any;
  constructor(viewUtils:import3.ViewUtils,parentView:import2.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import16.ViewContainer) {
    super(View_DayViewComponent29,renderType_DayViewComponent,import6.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import1.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._map_8 = import3.pureProxy1((p0:any):{[key: string]:any} => {
      return {'calendar-event-wrap': p0};
    });
  }
  createInternal(rootSelector:string):import7.ComponentRef<any> {
    this._el_0 = import3.createRenderElement(this.renderer,(null as any),'div',import3.EMPTY_INLINE_ARRAY,(null as any));
    this._NgClass_0_3 = new import27.Wrapper_NgClass(this.parentView.parentView.parentView.parentView.injectorGet(import30.IterableDiffers,this.parentView.parentView.parentView.parentIndex),this.parentView.parentView.parentView.parentView.injectorGet(import31.KeyValueDiffers,this.parentView.parentView.parentView.parentIndex),new import23.ElementRef(this._el_0),this.renderer);
    this._text_1 = this.renderer.createText(this._el_0,'\n                                    ',(null as any));
    this._anchor_2 = this.renderer.createTemplateAnchor(this._el_0,(null as any));
    this._vc_2 = new import16.ViewContainer(2,0,this,this._anchor_2);
    this._TemplateRef_2_5 = new import24.TemplateRef_(this,2,this._anchor_2);
    this._NgFor_2_6 = new import29.Wrapper_NgFor(this._vc_2.vcRef,this._TemplateRef_2_5,this.parentView.parentView.parentView.parentView.injectorGet(import30.IterableDiffers,this.parentView.parentView.parentView.parentIndex),this.parentView.parentView.parentView.ref);
    this._text_3 = this.renderer.createText(this._el_0,'\n                                ',(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._anchor_2,
      this._text_3
    ]
    ),(null as any));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import24.TemplateRef) && (2 === requestNodeIndex))) { return this._TemplateRef_2_5; }
    if (((token === import32.NgFor) && (2 === requestNodeIndex))) { return this._NgFor_2_6.context; }
    if (((token === import33.NgClass) && ((0 <= requestNodeIndex) && (requestNodeIndex <= 3)))) { return this._NgClass_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_0_0_0:any = this._map_8(this.parentView.context.$implicit.events);
    this._NgClass_0_3.check_ngClass(currVal_0_0_0,throwOnChange,false);
    this._NgClass_0_3.ngDoCheck(this,this._el_0,throwOnChange);
    const currVal_2_0_0:any = this.parentView.context.$implicit.events;
    this._NgFor_2_6.check_ngForOf(currVal_2_0_0,throwOnChange,false);
    this._NgFor_2_6.ngDoCheck(this,this._anchor_2,throwOnChange);
    this._vc_2.detectChangesInNestedViews(throwOnChange);
  }
  destroyInternal():void {
    this._vc_2.destroyNestedViews();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  createEmbeddedViewInternal(nodeIndex:number):import2.AppView<any> {
    if ((nodeIndex == 2)) { return new View_DayViewComponent30(this.viewUtils,this,2,this._anchor_2,this._vc_2); }
    return (null as any);
  }
}
class View_DayViewComponent30 extends import2.AppView<any> {
  _el_0:any;
  _NgStyle_0_3:import28.Wrapper_NgStyle;
  _text_1:any;
  _anchor_2:any;
  /*private*/ _vc_2:import16.ViewContainer;
  _TemplateRef_2_5:any;
  _NgTemplateOutlet_2_6:import35.Wrapper_NgTemplateOutlet;
  _text_3:any;
  _map_8:any;
  _map_9:any;
  constructor(viewUtils:import3.ViewUtils,parentView:import2.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import16.ViewContainer) {
    super(View_DayViewComponent30,renderType_DayViewComponent,import6.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import1.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._map_8 = import3.pureProxy4((p0:any,p1:any,p2:any,p3:any):{[key: string]:any} => {
      return {
        top: p0,
        left: p1,
        width: p2,
        height: p3
      }
      ;
    });
    this._map_9 = import3.pureProxy1((p0:any):{[key: string]:any} => {
      return {displayEvent: p0};
    });
  }
  createInternal(rootSelector:string):import7.ComponentRef<any> {
    this._el_0 = import3.createRenderElement(this.renderer,(null as any),'div',new import3.InlineArray4(4,'class','calendar-event','tappable',''),(null as any));
    this._NgStyle_0_3 = new import28.Wrapper_NgStyle(this.parentView.parentView.parentView.parentView.parentView.injectorGet(import31.KeyValueDiffers,this.parentView.parentView.parentView.parentView.parentIndex),new import23.ElementRef(this._el_0),this.renderer);
    this._text_1 = this.renderer.createText(this._el_0,'\n                                         ',(null as any));
    this._anchor_2 = this.renderer.createTemplateAnchor(this._el_0,(null as any));
    this._vc_2 = new import16.ViewContainer(2,0,this,this._anchor_2);
    this._TemplateRef_2_5 = new import24.TemplateRef_(this,2,this._anchor_2);
    this._NgTemplateOutlet_2_6 = new import35.Wrapper_NgTemplateOutlet(this._vc_2.vcRef);
    this._text_3 = this.renderer.createText(this._el_0,'\n                                    ',(null as any));
    var disposable_0:Function = import3.subscribeToRenderElement(this,this._el_0,new import3.InlineArray2(2,'click',(null as any)),this.eventHandler(this.handleEvent_0));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._anchor_2,
      this._text_3
    ]
    ),[disposable_0]);
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import24.TemplateRef) && (2 === requestNodeIndex))) { return this._TemplateRef_2_5; }
    if (((token === import36.NgTemplateOutlet) && (2 === requestNodeIndex))) { return this._NgTemplateOutlet_2_6.context; }
    if (((token === import34.NgStyle) && ((0 <= requestNodeIndex) && (requestNodeIndex <= 3)))) { return this._NgStyle_0_3.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_0_0_0:any = this._map_8((((37 * this.context.$implicit.startOffset) / this.parentView.parentView.parentView.parentView.context.hourParts) + 'px'),(((100 / this.context.$implicit.overlapNumber) * this.context.$implicit.position) + '%'),((100 / this.context.$implicit.overlapNumber) + '%'),((37 * ((this.context.$implicit.endIndex - this.context.$implicit.startIndex) - ((this.context.$implicit.endOffset + this.context.$implicit.startOffset) / this.parentView.parentView.parentView.parentView.context.hourParts))) + 'px'));
    this._NgStyle_0_3.check_ngStyle(currVal_0_0_0,throwOnChange,false);
    this._NgStyle_0_3.ngDoCheck(this,this._el_0,throwOnChange);
    const currVal_2_0_0:any = this._map_9(this.context.$implicit);
    this._NgTemplateOutlet_2_6.check_ngOutletContext(currVal_2_0_0,throwOnChange,false);
    const currVal_2_0_1:any = this.parentView.parentView.parentView.parentView.context.dayviewNormalEventTemplate;
    this._NgTemplateOutlet_2_6.check_ngTemplateOutlet(currVal_2_0_1,throwOnChange,false);
    this._NgTemplateOutlet_2_6.ngDoCheck(this,this._anchor_2,throwOnChange);
    this._vc_2.detectChangesInNestedViews(throwOnChange);
  }
  destroyInternal():void {
    this._vc_2.destroyNestedViews();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  createEmbeddedViewInternal(nodeIndex:number):import2.AppView<any> {
    if ((nodeIndex == 2)) { return new View_DayViewComponent31(this.viewUtils,this,2,this._anchor_2,this._vc_2); }
    return (null as any);
  }
  handleEvent_0(eventName:string,$event:any):boolean {
    this.markPathToRootAsCheckOnce();
    var result:boolean = true;
    if ((eventName == 'click')) {
      const pd_sub_0:any = ((<any>this.parentView.parentView.parentView.parentView.context.eventSelected(this.context.$implicit.event)) !== false);
      result = (pd_sub_0 && result);
    }
    return result;
  }
}
class View_DayViewComponent31 extends import2.AppView<any> {
  _text_0:any;
  constructor(viewUtils:import3.ViewUtils,parentView:import2.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import16.ViewContainer) {
    super(View_DayViewComponent31,renderType_DayViewComponent,import6.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import1.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
  }
  createInternal(rootSelector:string):import7.ComponentRef<any> {
    this._text_0 = this.renderer.createText((null as any),'\n                                         ',(null as any));
    this.init(this._text_0,((<any>this.renderer).directRenderer? (null as any): [this._text_0]),(null as any));
    return (null as any);
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._text_0,ctx);
  }
}
class View_DayViewComponent32 extends import2.AppView<any> {
  _el_0:any;
  _text_1:any;
  _el_2:any;
  _text_3:any;
  _anchor_4:any;
  /*private*/ _vc_4:import16.ViewContainer;
  _TemplateRef_4_5:any;
  _NgFor_4_6:import29.Wrapper_NgFor;
  _text_5:any;
  _text_6:any;
  constructor(viewUtils:import3.ViewUtils,parentView:import2.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import16.ViewContainer) {
    super(View_DayViewComponent32,renderType_DayViewComponent,import6.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import1.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
  }
  createInternal(rootSelector:string):import7.ComponentRef<any> {
    this._el_0 = import3.createRenderElement(this.renderer,(null as any),'table',new import3.InlineArray2(2,'class','table table-bordered table-fixed dayview-normal-event-table'),(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'\n                        ',(null as any));
    this._el_2 = import3.createRenderElement(this.renderer,this._el_0,'tbody',import3.EMPTY_INLINE_ARRAY,(null as any));
    this._text_3 = this.renderer.createText(this._el_2,'\n                        ',(null as any));
    this._anchor_4 = this.renderer.createTemplateAnchor(this._el_2,(null as any));
    this._vc_4 = new import16.ViewContainer(4,2,this,this._anchor_4);
    this._TemplateRef_4_5 = new import24.TemplateRef_(this,4,this._anchor_4);
    this._NgFor_4_6 = new import29.Wrapper_NgFor(this._vc_4.vcRef,this._TemplateRef_4_5,this.parentView.parentView.injectorGet(import30.IterableDiffers,this.parentView.parentIndex),this.parentView.ref);
    this._text_5 = this.renderer.createText(this._el_2,'\n                        ',(null as any));
    this._text_6 = this.renderer.createText(this._el_0,'\n                    ',(null as any));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._el_2,
      this._text_3,
      this._anchor_4,
      this._text_5,
      this._text_6
    ]
    ),(null as any));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import24.TemplateRef) && (4 === requestNodeIndex))) { return this._TemplateRef_4_5; }
    if (((token === import32.NgFor) && (4 === requestNodeIndex))) { return this._NgFor_4_6.context; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_4_0_0:any = this.parentView.context.views[2].rows;
    this._NgFor_4_6.check_ngForOf(currVal_4_0_0,throwOnChange,false);
    this._NgFor_4_6.ngDoCheck(this,this._anchor_4,throwOnChange);
    this._vc_4.detectChangesInNestedViews(throwOnChange);
  }
  destroyInternal():void {
    this._vc_4.destroyNestedViews();
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
  createEmbeddedViewInternal(nodeIndex:number):import2.AppView<any> {
    if ((nodeIndex == 4)) { return new View_DayViewComponent33(this.viewUtils,this,4,this._anchor_4,this._vc_4); }
    return (null as any);
  }
}
class View_DayViewComponent33 extends import2.AppView<any> {
  _el_0:any;
  _text_1:any;
  _el_2:any;
  _text_3:any;
  _text_4:any;
  _el_5:any;
  _text_6:any;
  _text_7:any;
  /*private*/ _expr_8:any;
  _pipe_date_0_5:any;
  constructor(viewUtils:import3.ViewUtils,parentView:import2.AppView<any>,parentIndex:number,parentElement:any,declaredViewContainer:import16.ViewContainer) {
    super(View_DayViewComponent33,renderType_DayViewComponent,import6.ViewType.EMBEDDED,viewUtils,parentView,parentIndex,parentElement,import1.ChangeDetectorStatus.CheckAlways,declaredViewContainer);
    this._expr_8 = import1.UNINITIALIZED;
  }
  createInternal(rootSelector:string):import7.ComponentRef<any> {
    this._el_0 = import3.createRenderElement(this.renderer,(null as any),'tr',import3.EMPTY_INLINE_ARRAY,(null as any));
    this._text_1 = this.renderer.createText(this._el_0,'\n                            ',(null as any));
    this._el_2 = import3.createRenderElement(this.renderer,this._el_0,'td',new import3.InlineArray2(2,'class','calendar-hour-column text-center'),(null as any));
    this._text_3 = this.renderer.createText(this._el_2,'',(null as any));
    this._text_4 = this.renderer.createText(this._el_0,'\n                            ',(null as any));
    this._el_5 = import3.createRenderElement(this.renderer,this._el_0,'td',new import3.InlineArray2(2,'class','calendar-cell'),(null as any));
    this._text_6 = this.renderer.createText(this._el_5,'\n                            ',(null as any));
    this._text_7 = this.renderer.createText(this._el_0,'\n                        ',(null as any));
    this._pipe_date_0_5 = import3.pureProxy2((<View_DayViewComponent0>this.parentView.parentView)._pipe_date_0.transform.bind((<View_DayViewComponent0>this.parentView.parentView)._pipe_date_0));
    this.init(this._el_0,((<any>this.renderer).directRenderer? (null as any): [
      this._el_0,
      this._text_1,
      this._el_2,
      this._text_3,
      this._text_4,
      this._el_5,
      this._text_6,
      this._text_7
    ]
    ),(null as any));
    return (null as any);
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const valUnwrapper:any = new import1.ValueUnwrapper();
    valUnwrapper.reset();
    const currVal_8:any = import3.inlineInterpolate(1,'\n                                ',valUnwrapper.unwrap(import3.castByValue(this._pipe_date_0_5,(<View_DayViewComponent0>this.parentView.parentView)._pipe_date_0.transform)(this.context.$implicit.time,this.parentView.parentView.context.formatHourColumn)),'\n                            ');
    if ((valUnwrapper.hasWrappedValue || import3.checkBinding(throwOnChange,this._expr_8,currVal_8))) {
      this.renderer.setText(this._text_3,currVal_8);
      this._expr_8 = currVal_8;
    }
  }
  visitRootNodesInternal(cb:any,ctx:any):void {
    cb(this._el_0,ctx);
  }
}
